{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## README: Making input fasta files\n",
    "\n",
    "This script wrangles multiple sequence alignments and fasta files from `Nextstrain/fauna` to make fasta files of sequences that we want to include specifically in our analysis of Zika in the Americas. A variety of fasta files are outputted, including ones that can be read in to BEAST, and ones that can be read in to `Nextstrain/augur`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### import libraries ####\n",
    "from Bio import SeqIO\n",
    "from Bio import AlignIO\n",
    "import datetime\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import csv\n",
    "date = datetime.datetime.now().strftime (\"%Y-%m-%d\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Remove sequences from geographic areas we do not want to include (this is everything other than Americas and French Polynesia)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "738 sequences were available from Fauna.\n",
      "568 sequences meet the geographic criteria.\n"
     ]
    }
   ],
   "source": [
    "fauna_file = \"../data/zika-fauna-with-correct-usvi-seqs-2019-11-20.fasta\"\n",
    "fauna_dict = SeqIO.to_dict(SeqIO.parse(fauna_file, 'fasta'))\n",
    "print('{} sequences were available from Fauna.'.format(len(fauna_dict)))\n",
    "\n",
    "regions_to_exclude = ['southeast_asia', 'japan_korea', 'south_asia', 'africa', 'china','europe']\n",
    "reference_strain_name = 'PF13/251013_18' \n",
    "\n",
    "geoPruned_fauna_dict = {fauna_dict[key].description:fauna_dict[key].seq for key in fauna_dict.keys() if key.split('|')[4] not in regions_to_exclude}\n",
    "print ('{} sequences meet the geographic criteria.'.format(len(geoPruned_fauna_dict)))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Change region naming so that caribbean and central_america are not counted as \"north_america\" (which is fauna's default)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "central_america = ['panama', 'costa_rica', 'honduras', 'el_salvador', 'nicaragua', 'guatemala', 'belize']\n",
    "caribbean = ['barbados', 'usvi', 'puerto_rico', 'dominican_republic', 'martinique', 'guadeloupe', 'saint_barthelemy', 'jamaica', 'haiti', 'cuba']\n",
    "\n",
    "regionallyNamed_dict = {}\n",
    "\n",
    "for key in geoPruned_fauna_dict.keys():\n",
    "    country = key.split('|')[5]\n",
    "    if country in central_america:\n",
    "        new_key = key.replace('north_america', 'central_america')\n",
    "    elif country in caribbean:\n",
    "        new_key = key.replace('north_america', 'caribbean')\n",
    "    else:\n",
    "        new_key = key\n",
    "    regionallyNamed_dict[new_key] = geoPruned_fauna_dict[key]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 3: Remove sequences that lack sufficient numbers of informative bases to align well. \n",
    "\n",
    "Here I want sequences to have at least 80% of their bases be informative calls. I'm taking the sequences that were from the correct region (saved in the `regionallyNamed_dict`), selecting all USVI sequences and then any other sequence in that dictionary that is 80% or more informative base calls, and saving that trimmed down dataset to a new dictionary called `high_qual_seqs`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "## replace n's with gaps, and count n's in sequences (alignment checks)\n",
    "def proportion_n(sequence):\n",
    "    \"counts numbers of N's in a sequence to perform QC and see how many non-informative sites exist\"\n",
    "    counter = 0\n",
    "    denominator = float(len(sequence))\n",
    "    for base in sequence:\n",
    "        if base in ['N','n']:\n",
    "            counter +=1\n",
    "    return (counter/denominator)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 476 sequences that are sampled from a relevant region and have over 70% informative bases\n"
     ]
    }
   ],
   "source": [
    "high_qual_seqs = {}\n",
    "for key in regionallyNamed_dict.keys():\n",
    "    if key.split('|')[5] == 'usvi': #keep all USVI sequences.\n",
    "        high_qual_seqs[key] = regionallyNamed_dict[key]\n",
    "    else:\n",
    "        n_fraction = proportion_n(regionallyNamed_dict[key])\n",
    "        if n_fraction < 0.2:\n",
    "            high_qual_seqs[key] = regionallyNamed_dict[key]\n",
    "\n",
    "print(\"There are {} sequences that are sampled from a relevant region and have over 70% informative bases\".format(len(high_qual_seqs.keys())))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 4: Remove any sequences for which we do not have permission to include in published analyses.\n",
    "\n",
    "From the set of sequences that meet my quality and region thresholds I need to filter out any that are in genbank, but that were unpublished that that we don't have permission to include. The input dictionary here is `high_qual_seqs` and the output dictionary is `publishable_seqs_dict`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first step, print out all of the accessions and authors for the sequences that are still left in the preliminary dataset!\n",
    "# I'll then go through this list and look for which sequences have been published on, which are in genbank but unpublished,\n",
    "# And that can guide \n",
    "with open('../data/sequence-accessions-author-info.csv', 'w') as file:\n",
    "    for key in high_qual_seqs.keys():\n",
    "        file.write(\"%s,%s,%s\\n\"%(key.split('|')[0], key.split('|')[2], key.split('|')[10]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 394 genomes that we can include in published analyses.\n"
     ]
    }
   ],
   "source": [
    "# read in the dataframe that has the permissions information,\n",
    "# then parse that to select out all strains that can be included in a publishable analysis\n",
    "# these are the strains that should be used, and form the fauna subset we want.\n",
    "genome_permissions = pd.read_csv('../data/sequence-permissions-info-manually-verified.csv', delimiter =',')\n",
    "\n",
    "publishable_strains = []\n",
    "for i in range(len(genome_permissions)):\n",
    "    record = genome_permissions.iloc[i]\n",
    "    if record['preliminarily_include'] == 'yes':\n",
    "        publishable_strains.append(record['strain_name'])\n",
    "\n",
    "# using the strains in the publishable_strains list, pull out the full fauna headers (and sequences)\n",
    "# for each strain that can be published on.\n",
    "# also making sure we are only dealing with sequences within the countries of interest (Americans and Oceania)\n",
    "publishable_seqs_dict = {}\n",
    "for strain in publishable_strains:\n",
    "    for key in high_qual_seqs.keys():\n",
    "        if key.startswith(strain):\n",
    "            publishable_seqs_dict[key] = high_qual_seqs[key]\n",
    "            \n",
    "print(\"There are {} genomes that we can include in published analyses.\".format(len(publishable_seqs_dict.keys())))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 5: Remove known sequences which are all cell culture passages of the same clinical isolate\n",
    "\n",
    "There is a large number of Colombian sequences that are all sequences of various passages of a single strain. Here, I want to keep the one sequence that was directly sequenced from the clinical isolate, and remove all of the other sequences. The output dictionary here (with the sequences I want to keep) is `publishable_seqs_no_cell_culture_passage_dict`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There were 33 sequences from different culture passages of COL/FLR/2015 that should be removed.\n",
      "There are now 361 sequences that should be included in the build.\n"
     ]
    }
   ],
   "source": [
    "col_flr_strains_to_remove = [strain for strain in publishable_strains if strain.startswith(\"COL/FLR\") and strain != 'COL/FLR/2015']\n",
    "print(\"There were {} sequences from different culture passages of COL/FLR/2015 that should be removed.\".format(len(col_flr_strains_to_remove)))\n",
    "\n",
    "publishable_seqs_no_cell_culture_passage_dict = {}\n",
    "for key in publishable_seqs_dict.keys():\n",
    "    if key.split('|')[0] not in col_flr_strains_to_remove:\n",
    "        publishable_seqs_no_cell_culture_passage_dict[key] = publishable_seqs_dict[key]\n",
    "\n",
    "assert(len(publishable_seqs_no_cell_culture_passage_dict.keys()) == (len(publishable_seqs_dict.keys()) -(len(col_flr_strains_to_remove))))\n",
    "print(\"There are now {} sequences that should be included in the build.\".format(len(publishable_seqs_no_cell_culture_passage_dict)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now, write out the sequences that should be part of the build to an input fasta."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"../data/usvi-and-high-qual-americas-input-data.fasta\",\"w\") as file:\n",
    "    for key in publishable_seqs_no_cell_culture_passage_dict.keys():\n",
    "        file.write(str(\">\" + key + \"\\n\" + publishable_seqs_no_cell_culture_passage_dict[key].lower() + \"\\n\" ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2016-08-03\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "from pprint import pprint\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/augur/builds/zika/prepared/zika.json') as zika_json_file:    \n",
    "    zika_json_data = json.load(zika_json_file)\n",
    "#pprint(data)\n",
    "#print data.keys()\n",
    "print(zika_json_data['sequences']['USVI/45/2016']['attributes']['raw_date'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usvi\n"
     ]
    }
   ],
   "source": [
    "print(zika_json_data['sequences']['USVI/45/2016']['attributes']['country'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quick map of how the JSON will get parsed!\n",
    "\n",
    "Layer 1: `data.keys()`info, colors, lat_longs, reference, sequences (things we care about here are sequences)\n",
    "\n",
    "Layer 2: `data[sequences].keys()`: Brings up all strains, with canonicized strain names ( e.g. USVI/45/2016).\n",
    "\n",
    "Layer 3: `data[sequences][<strain>].keys()`: Brings up attributes and seq. The sequences is the full, unaligned sequence, attributes have further keys.\n",
    "\n",
    "Layer 4: `data[sequences][<strain>][attributes].keys()`: Brings up division, title, url, country, regio, db, accession, raw_date, strain, authors, paper_url, and journal."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/alliblk/Desktop/gitrepos/augur/builds/zika/processed/zika_aligned_stripped.mfa\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### infile paths #### \n",
    "zika_msa_stripped = \"/Users/alliblk/Desktop/gitrepos/augur/builds/zika/processed/zika_aligned_stripped.mfa\"\n",
    "fauna_file = \"/Users/alliblk/Desktop/gitrepos/fauna/data/zika.fasta\"\n",
    "\n",
    "#### outfile paths #### \n",
    "americas_file = '/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/american-zika-{}.fasta'.format(date)\n",
    "americas_frenchpol_file = '/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/american-frenchPolyn-zika-{}.fasta'.format(date)\n",
    "\n",
    "usvi_file = \"/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi-{}.fasta\".format(date)\n",
    "usvi_primary_clade_file = '/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi-primary-clade-{}.fasta'.format(date)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step 1: Load in all available sequences in from Fauna. This should be all Zika genomes that have been made publicly available on GenBank, ViPR, and GitHub. Note that this does NOT mean that all of the genomes have been published on.\n",
    "\n",
    "Note that you will need to clone the `nextstrain/fauna` git repo and run fauna to have this file. \n",
    "\n",
    "Download zika sequences using the command call in the `builds/ZIKA.README` file:\n",
    "\n",
    "`python vdb/zika_download.py -db vdb -v zika --fstem zika --resolve_method choose_genbank`\n",
    "\n",
    "This is important because the reference strain PF13/251013_18 that we're using as our outgroup has been sequenced twice. The version of that sequence that we want is Troesemeier et al's. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step 2: Make a subset of the fauna sequences that only includes the geographic regions of interest. Here, I want only Zika sequences from the Americas. I need an outgroup sequence to properly root the tree however, so I'm also going to include a single genome from French Polynesia. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "old_permission_info = pd.read_csv('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/sequences_and_permissions.txt',delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print [key.split('|')[2] for key in geoPruned_fauna_dict.keys()]\n",
    "new_accessions_to_check = [key for key in geoPruned_fauna_dict.keys() if key.split('|')[2] not in pd.Series(old_permission_info['accession_number']).tolist()]\n",
    "\n",
    "strain_names = [accession.split('|')[0] for accession in new_accessions_to_check]\n",
    "ncbi_id = [accession.split('|')[2] for accession in new_accessions_to_check]\n",
    "lead_author =  [accession.split('|')[10].replace('et al','') for accession in new_accessions_to_check]\n",
    "\n",
    "\n",
    "new_accessions_df = pd.DataFrame(np.column_stack([strain_names, ncbi_id, lead_author]), columns= ['strain_name', 'accession_number','lead_author'])\n",
    "new_accessions_df.to_csv('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/new_accesions_between_08152017_and_03212018.csv', index=False)\n",
    "#new_accessions_df = pd.DataFrame(strain_names, ncbi_id, lead_author, columns = ['strain_name', 'accession_number','lead_author','permission_type','publication_pubmed_id','permission_to_use','preliminarily_include'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31\n",
      "[1, 2, 3, 4, 5, 6, 7, 11, 12, 13, 19, 20, 21, 22, 23, 24, 25, 27, 28, 30, 32, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43]\n"
     ]
    }
   ],
   "source": [
    "VIseqs = [key.split('|')[0] for key in geoPruned_fauna_dict.keys() if key.split('|')[0].startswith('USVI') or key.split('|')[0].startswith('USIV')]\n",
    "print len(VIseqs)  \n",
    "ordered_VI_strain_numbers = sorted([int(strain.split('/')[1]) for strain in VIseqs])\n",
    "print ordered_VI_strain_numbers\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step 3: Filter sequences so that we only include sequences that we are allowed to publish on. These include any sequences that have been previously released during publication, or any sequences for which the authors have given us permission to use them in our analysis.\n",
    "\n",
    "This information has been collated in a table that I'm going to import in here. Note that the table is NOT exhaustive for all genomes available from Fauna. Rather, it only includes sequences from the Americas. Sequence accessions have been manually checked and matched with publications if they are published on (with pubmed ID information). If not published on, sequences have a flag that denotes whether author permission was received or not.\n",
    "\n",
    "I have an indicator in this table as well that is labeled `premliminarily_include`. If a record says `no` in this column, it is because it needed permissions to be requested, but would likely not add more to the analysis, and therefore was deemed not necessary to pursue permissions on.\n",
    "\n",
    "Below I have lists of all the publications as they're listed in the Fauna headers, which represent the title given in GenBank but _not necessarily the title of the actual published manuscript_. They are divided into publications that are published, ones that are not published, and ones that are in fact published but the actual manuscript name is different than the GenBank publication name, and therefore they appear to be unpublished in GenBank.\n",
    "\n",
    "---------------\n",
    "\n",
    "`published_submissions` = ['First Complete Genome Sequence of Zika Virus (Flaviviridae, Flavivirus) from an Autochthonous Transmission in Brazil',\n",
    "'Genetic characterization of the Zika virus epidemic in the US Virgin Islands',\n",
    "'Isolation of Infective Zika Virus from Urine and Saliva of Patients in Brazil',\n",
    "'Full-length infectious cDNA clone of Zika virus from 2015 epidemic in Brazil: development and characterization of recombinant viruses in cell lines from human placenta, testis, and brain',\n",
    "'Zika virus complete genome from Salvador, Bahia, Brazil',\n",
    "'Complete Genome Sequences of Three Historically Important, Spatiotemporally Distinct, and Genetically Divergent Strains of Zika Virus: MR-766, P6-740, and PRVABC-59',\n",
    "'Isolation of infectious Zika virus from saliva and prolonged viral RNA shedding in a traveller returning from the Dominican Republic to Italy, January 2016',\n",
    "'Fatal Zika Virus Infection with Secondary Nonsexual Transmission',\n",
    "'Quasispecies composition and evolution of a typical Zika virus clinical isolate from Suriname',\n",
    "'First Complete Genome Sequences of Zika Virus Isolated from Febrile Patient Sera in Ecuador',\n",
    "'Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas',\n",
    "'Genomic epidemiology reveals multiple introductions of Zika virus into the United States',\n",
    "'Coinfection With Zika and Dengue-2 Viruses in a Traveler Returning From Haiti, 2016: Clinical Presentation and Genetic Analysis',\n",
    "'Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples',\n",
    "'Large returning population of oversea Chinese of Guangdong province combined with tourists increase the risks of ZIKV transmission in Guangdong, China',\n",
    "'Zika Virus Targets Different Primary Human Placental Cells, Suggesting Two Routes for Vertical Transmission',\n",
    "'Establishment and cryptic transmission of Zika virus in Brazil and the Americas',\n",
    "'Zika Virus Outbreak in Haiti in 2014: Molecular and Clinical Data',\n",
    "'Prolonged Detection of Zika Virus in Vaginal Secretions and Whole Blood',\n",
    "'Complete Genome Sequences of Identical Zika virus Isolates in a Nursing Mother and Her Infant',\n",
    "'?',\n",
    "'Zika virus in the Americas: early epidemiological and genetic findings',\n",
    "'Full Genome Sequence and sfRNA Interferon Antagonist Activity of Zika Virus from Recife, Brazil',\n",
    "'Distinct Zika Virus Lineage in Salvador, Bahia, Brazil',\n",
    "'Genome sequence of a candidate World Health Organization reference strain for Zika virus for nucleic acid testing',\n",
    "'Zika virus evolution and spread in the Americas',\n",
    "'Zika virus genome from the Americas',\n",
    "'Complete coding sequence of Zika virus from Martinique outbreak in 2015',\n",
    "'Zika Virus Associated with Microcephaly',\n",
    "'Characterization of a Zika Virus Isolate from Colombia',\n",
    "'Detection and sequencing of Zika virus from amniotic fluid of fetuses with microcephaly in Brazil: a case study',\n",
    "'Phylogeny of Zika Virus in Western Hemisphere, 2015',\n",
    "'Complete genomic sequence of Zika virus isolated from a clinical semen sample',\n",
    "'Infection dynamics in a traveller with persistent shedding of Zika virus RNA in semen for six months after returning from Haiti to Italy, January 2016',                        \n",
    "                  ]\n",
    "\n",
    "`published_submissions_with_incorrect_publication_name` = [ 'Complete Zika Virus Genome Sequences from a Serum Sample and after Isolation in Vero Cells',\n",
    "'Full coding sequence of the Zika Virus isolate HS-2015-BA-01',\n",
    "'Large returning population of oversea Chinese of Guangdong province combined with tourists increase the risks of ZIKV transmission in Guangdong, China',\n",
    "'The isolation and characterization of Zika virus imported into Guangdong, China',\n",
    "'Full genomic characterization of a Zika virus isolate from Colombia',\n",
    "'Molecular characterization of Zika virus in Mexico',\n",
    "'Repurposed drug candidates to treat ZIKV infection in pregnancy',\n",
    "'Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach',\n",
    "'Zika virus infection with prolonged maternal viremia and fetal brain abnormalities',\n",
    "'Complete genome of Zika virus GZ01 isolate from China',\n",
    "'Complete genomic sequence of Zika virus isolated from a traveler to Colombia in 2016',\n",
    "                                                       \n",
    "                                                         \n",
    "]\n",
    "\n",
    "`unpublished_submissions` = [\n",
    "'Imported cases of Zika virus disease in Russia in 2016-2017',\n",
    "'Complete Genome of a Zika Virus Isolate, Amazonas, Brazil, 2016',\n",
    "'Imported cases of Zika virus disease in Russia in 2016-2017',\n",
    "'Zika viral sequence identified in Fortaleza, Brazil',\n",
    "'Comparative analysis between the genomes of intra-host and cell cultured Zika virus obtained from a Mexican symptomatic patient'\n",
    "'Zika virus from breast milk',\n",
    "'Genome sequences of Zika viruses isolated in Nicaragua',              \n",
    "'Emergence of Zika virus in Brazil: Detection of Asian genotype',\n",
    "'Brazilian Zika virus genome sequence',\n",
    "'Zika viruses in Haiti during an outbreak of Chikungunya Fever in mid 2014', #white et al sequences\n",
    "'Zika virus sequence from Cuba traveler diagnosed in the United States (Miami, FL)', #nate\n",
    "'Zika virus genomes from human cases in Florida, USA',\n",
    "'Isolation of infectious Zika virus from sera in a traveller returning from Brazil to Italy',\n",
    "'Zika virus infection in pregnant mice',\n",
    "'Zika virus polyprotein sequence isolated from a fetus-brain of Cuba pregnant traveler diagnosed in Spain (Madrid)',\n",
    "'Genome sequences of four Zika virus isolates from Brazil',\n",
    "'Direct Submission'\n",
    "                          ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain_name</th>\n",
       "      <th>accession_number</th>\n",
       "      <th>lead_author</th>\n",
       "      <th>permission_type</th>\n",
       "      <th>publication_pubmed_id</th>\n",
       "      <th>permission_to_use</th>\n",
       "      <th>preliminarily_include</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PHE_Guadeloupe</td>\n",
       "      <td>KX673530</td>\n",
       "      <td>Atkinson</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27738033.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MEX_I_7</td>\n",
       "      <td>KX247632</td>\n",
       "      <td>Barrows</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27476412.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Dominican_Republic/2016/PD1</td>\n",
       "      <td>KU853012</td>\n",
       "      <td>Barzon</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>26987769.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Haiti/2016/PD</td>\n",
       "      <td>KX269878</td>\n",
       "      <td>Barzon</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27542178.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VEN/UF_1/2016</td>\n",
       "      <td>KX702400</td>\n",
       "      <td>Blohm</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28450510.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VEN/UF_2/2016</td>\n",
       "      <td>KX893855</td>\n",
       "      <td>Blohm</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28450510.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Rio_U1</td>\n",
       "      <td>KU926309</td>\n",
       "      <td>Bonaldo</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27341420.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Rio_S1</td>\n",
       "      <td>KU926310</td>\n",
       "      <td>Bonaldo</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27341420.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MEX_CIENI551</td>\n",
       "      <td>KY120348</td>\n",
       "      <td>Boukadida</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28336600.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Brazil_ZKV2015</td>\n",
       "      <td>KU497555</td>\n",
       "      <td>Calvet</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>26897108.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>COL/UF_1/2016</td>\n",
       "      <td>KX247646</td>\n",
       "      <td>Cherabuddi</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28348794.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SPH2015</td>\n",
       "      <td>KU321639</td>\n",
       "      <td>Cunha</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>26941134.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MEX/InDRE/2016</td>\n",
       "      <td>KU922960</td>\n",
       "      <td>Diaz-quinonez</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27491989.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Brazil/PE243/2015</td>\n",
       "      <td>KX197192</td>\n",
       "      <td>Donald</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27706161.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FB_GWUH_2016</td>\n",
       "      <td>KU870645</td>\n",
       "      <td>Driggers</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>27028667.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Z1106033</td>\n",
       "      <td>KU312312</td>\n",
       "      <td>Enfissi</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>26775124.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Brazil/2015/ZBRA105</td>\n",
       "      <td>KY558989</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Brazil/2016/ZBRC14</td>\n",
       "      <td>KY558990</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Brazil/2016/ZBRC16</td>\n",
       "      <td>KY558991</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Brazil/2016/ZBRC18</td>\n",
       "      <td>KY558992</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Brazil/2016/ZBRC25</td>\n",
       "      <td>KY558993</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Brazil/2016/ZBRC28</td>\n",
       "      <td>KY558994</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Brazil/2015/ZBRC301</td>\n",
       "      <td>KY558995</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Brazil/2015/ZBRC302</td>\n",
       "      <td>KY558996</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Brazil/2015/ZBRC303</td>\n",
       "      <td>KY558997</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Brazil/2015/ZBRC313</td>\n",
       "      <td>KY558998</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Brazil/2016/ZBRC319</td>\n",
       "      <td>KY558999</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Brazil/2015/ZBRC321</td>\n",
       "      <td>KY559000</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Brazil/2015/ZBRD103</td>\n",
       "      <td>KY559001</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Brazil/2015/ZBRD107</td>\n",
       "      <td>KY559002</td>\n",
       "      <td>Faria</td>\n",
       "      <td>previously_published</td>\n",
       "      <td>28538727.0</td>\n",
       "      <td>not_necessary</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>MEX/2016/mex33</td>\n",
       "      <td>MF801411</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>MEX/2016/mex34</td>\n",
       "      <td>MF801412</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>MEX/2016/mex35</td>\n",
       "      <td>MF801413</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>MEX/2016/mex36</td>\n",
       "      <td>MF801414</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>MEX/2016/mex37</td>\n",
       "      <td>MF801415</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>MEX/2016/mex38</td>\n",
       "      <td>MF801416</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>MEX/2016/mex40</td>\n",
       "      <td>MF801417</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>MEX/2016/mex42</td>\n",
       "      <td>MF801418</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>MEX/2016/mex43</td>\n",
       "      <td>MF801419</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>MEX/2016/mex44</td>\n",
       "      <td>MF801420</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>MEX/2016/mex45</td>\n",
       "      <td>MF801421</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>MEX/2016/mex46</td>\n",
       "      <td>MF801422</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>MEX/2016/mex47</td>\n",
       "      <td>MF801423</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>MEX/2016/mex50</td>\n",
       "      <td>MF801424</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>NIC/2016/Nicaragua_3358</td>\n",
       "      <td>MF801425</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>NIC/2016/Nicaragua_3912</td>\n",
       "      <td>MF801426</td>\n",
       "      <td>Theze</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>GD01</td>\n",
       "      <td>KU740184</td>\n",
       "      <td>Tong</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>Mexico/INPER38b/2016</td>\n",
       "      <td>MG494697</td>\n",
       "      <td>Valdespino-Vazquez</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>CX5</td>\n",
       "      <td>KX986760</td>\n",
       "      <td>Wallau</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>CX17</td>\n",
       "      <td>KX986761</td>\n",
       "      <td>Wallau</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>Haiti/0029/2014</td>\n",
       "      <td>KY415986</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>Haiti/0033/2014</td>\n",
       "      <td>KY415987</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>Haiti/0054/2014</td>\n",
       "      <td>KY415988</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>Haiti/0074/2014</td>\n",
       "      <td>KY415989</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>Haiti/0036/2014</td>\n",
       "      <td>KY415990</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>Haiti/0097/2014</td>\n",
       "      <td>KY415991</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>mosquito/Haiti/1682/2016</td>\n",
       "      <td>MF384325</td>\n",
       "      <td>White</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>EC/Esmeraldas/121/2016</td>\n",
       "      <td>MF794971</td>\n",
       "      <td>Wise</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>GDZ16001</td>\n",
       "      <td>KU761564</td>\n",
       "      <td>Wu</td>\n",
       "      <td>unpublished</td>\n",
       "      <td>NaN</td>\n",
       "      <td>permission_not_received</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     strain_name accession_number          lead_author  \\\n",
       "0                 PHE_Guadeloupe         KX673530             Atkinson   \n",
       "1                        MEX_I_7         KX247632              Barrows   \n",
       "2    Dominican_Republic/2016/PD1         KU853012               Barzon   \n",
       "3                  Haiti/2016/PD         KX269878               Barzon   \n",
       "4                  VEN/UF_1/2016         KX702400                Blohm   \n",
       "5                  VEN/UF_2/2016         KX893855                Blohm   \n",
       "6                         Rio_U1         KU926309              Bonaldo   \n",
       "7                         Rio_S1         KU926310              Bonaldo   \n",
       "8                   MEX_CIENI551         KY120348            Boukadida   \n",
       "9                 Brazil_ZKV2015         KU497555               Calvet   \n",
       "10                 COL/UF_1/2016         KX247646           Cherabuddi   \n",
       "11                       SPH2015         KU321639                Cunha   \n",
       "12                MEX/InDRE/2016         KU922960        Diaz-quinonez   \n",
       "13             Brazil/PE243/2015         KX197192               Donald   \n",
       "14                  FB_GWUH_2016         KU870645             Driggers   \n",
       "15                      Z1106033         KU312312              Enfissi   \n",
       "16           Brazil/2015/ZBRA105         KY558989                Faria   \n",
       "17            Brazil/2016/ZBRC14         KY558990                Faria   \n",
       "18            Brazil/2016/ZBRC16         KY558991                Faria   \n",
       "19            Brazil/2016/ZBRC18         KY558992                Faria   \n",
       "20            Brazil/2016/ZBRC25         KY558993                Faria   \n",
       "21            Brazil/2016/ZBRC28         KY558994                Faria   \n",
       "22           Brazil/2015/ZBRC301         KY558995                Faria   \n",
       "23           Brazil/2015/ZBRC302         KY558996                Faria   \n",
       "24           Brazil/2015/ZBRC303         KY558997                Faria   \n",
       "25           Brazil/2015/ZBRC313         KY558998                Faria   \n",
       "26           Brazil/2016/ZBRC319         KY558999                Faria   \n",
       "27           Brazil/2015/ZBRC321         KY559000                Faria   \n",
       "28           Brazil/2015/ZBRD103         KY559001                Faria   \n",
       "29           Brazil/2015/ZBRD107         KY559002                Faria   \n",
       "..                           ...              ...                  ...   \n",
       "407               MEX/2016/mex33         MF801411               Theze    \n",
       "408               MEX/2016/mex34         MF801412               Theze    \n",
       "409               MEX/2016/mex35         MF801413               Theze    \n",
       "410               MEX/2016/mex36         MF801414               Theze    \n",
       "411               MEX/2016/mex37         MF801415               Theze    \n",
       "412               MEX/2016/mex38         MF801416               Theze    \n",
       "413               MEX/2016/mex40         MF801417               Theze    \n",
       "414               MEX/2016/mex42         MF801418               Theze    \n",
       "415               MEX/2016/mex43         MF801419               Theze    \n",
       "416               MEX/2016/mex44         MF801420               Theze    \n",
       "417               MEX/2016/mex45         MF801421               Theze    \n",
       "418               MEX/2016/mex46         MF801422               Theze    \n",
       "419               MEX/2016/mex47         MF801423               Theze    \n",
       "420               MEX/2016/mex50         MF801424               Theze    \n",
       "421      NIC/2016/Nicaragua_3358         MF801425               Theze    \n",
       "422      NIC/2016/Nicaragua_3912         MF801426               Theze    \n",
       "423                         GD01         KU740184                Tong    \n",
       "424         Mexico/INPER38b/2016         MG494697  Valdespino-Vazquez    \n",
       "425                          CX5         KX986760              Wallau    \n",
       "426                         CX17         KX986761              Wallau    \n",
       "427              Haiti/0029/2014         KY415986               White    \n",
       "428              Haiti/0033/2014         KY415987               White    \n",
       "429              Haiti/0054/2014         KY415988               White    \n",
       "430              Haiti/0074/2014         KY415989               White    \n",
       "431              Haiti/0036/2014         KY415990               White    \n",
       "432              Haiti/0097/2014         KY415991               White    \n",
       "433     mosquito/Haiti/1682/2016         MF384325               White    \n",
       "434       EC/Esmeraldas/121/2016         MF794971                Wise    \n",
       "435                     GDZ16001         KU761564                  Wu    \n",
       "436                          NaN              NaN                  NaN   \n",
       "\n",
       "          permission_type  publication_pubmed_id        permission_to_use  \\\n",
       "0    previously_published             27738033.0            not_necessary   \n",
       "1    previously_published             27476412.0            not_necessary   \n",
       "2    previously_published             26987769.0            not_necessary   \n",
       "3    previously_published             27542178.0            not_necessary   \n",
       "4    previously_published             28450510.0            not_necessary   \n",
       "5    previously_published             28450510.0            not_necessary   \n",
       "6    previously_published             27341420.0            not_necessary   \n",
       "7    previously_published             27341420.0            not_necessary   \n",
       "8    previously_published             28336600.0            not_necessary   \n",
       "9    previously_published             26897108.0            not_necessary   \n",
       "10   previously_published             28348794.0            not_necessary   \n",
       "11   previously_published             26941134.0            not_necessary   \n",
       "12   previously_published             27491989.0            not_necessary   \n",
       "13   previously_published             27706161.0            not_necessary   \n",
       "14   previously_published             27028667.0            not_necessary   \n",
       "15   previously_published             26775124.0            not_necessary   \n",
       "16   previously_published             28538727.0            not_necessary   \n",
       "17   previously_published             28538727.0            not_necessary   \n",
       "18   previously_published             28538727.0            not_necessary   \n",
       "19   previously_published             28538727.0            not_necessary   \n",
       "20   previously_published             28538727.0            not_necessary   \n",
       "21   previously_published             28538727.0            not_necessary   \n",
       "22   previously_published             28538727.0            not_necessary   \n",
       "23   previously_published             28538727.0            not_necessary   \n",
       "24   previously_published             28538727.0            not_necessary   \n",
       "25   previously_published             28538727.0            not_necessary   \n",
       "26   previously_published             28538727.0            not_necessary   \n",
       "27   previously_published             28538727.0            not_necessary   \n",
       "28   previously_published             28538727.0            not_necessary   \n",
       "29   previously_published             28538727.0            not_necessary   \n",
       "..                    ...                    ...                      ...   \n",
       "407           unpublished                    NaN  permission_not_received   \n",
       "408           unpublished                    NaN  permission_not_received   \n",
       "409           unpublished                    NaN  permission_not_received   \n",
       "410           unpublished                    NaN  permission_not_received   \n",
       "411           unpublished                    NaN  permission_not_received   \n",
       "412           unpublished                    NaN  permission_not_received   \n",
       "413           unpublished                    NaN  permission_not_received   \n",
       "414           unpublished                    NaN  permission_not_received   \n",
       "415           unpublished                    NaN  permission_not_received   \n",
       "416           unpublished                    NaN  permission_not_received   \n",
       "417           unpublished                    NaN  permission_not_received   \n",
       "418           unpublished                    NaN  permission_not_received   \n",
       "419           unpublished                    NaN  permission_not_received   \n",
       "420           unpublished                    NaN  permission_not_received   \n",
       "421           unpublished                    NaN  permission_not_received   \n",
       "422           unpublished                    NaN  permission_not_received   \n",
       "423           unpublished                    NaN  permission_not_received   \n",
       "424           unpublished                    NaN  permission_not_received   \n",
       "425           unpublished                    NaN  permission_not_received   \n",
       "426           unpublished                    NaN  permission_not_received   \n",
       "427           unpublished                    NaN  permission_not_received   \n",
       "428           unpublished                    NaN  permission_not_received   \n",
       "429           unpublished                    NaN  permission_not_received   \n",
       "430           unpublished                    NaN  permission_not_received   \n",
       "431           unpublished                    NaN  permission_not_received   \n",
       "432           unpublished                    NaN  permission_not_received   \n",
       "433           unpublished                    NaN  permission_not_received   \n",
       "434           unpublished                    NaN  permission_not_received   \n",
       "435           unpublished                    NaN  permission_not_received   \n",
       "436                   NaN                    NaN                      NaN   \n",
       "\n",
       "    preliminarily_include  Unnamed: 7  \n",
       "0                     yes         NaN  \n",
       "1                     yes         NaN  \n",
       "2                     yes         NaN  \n",
       "3                     yes         NaN  \n",
       "4                     yes         NaN  \n",
       "5                     yes         NaN  \n",
       "6                     yes         NaN  \n",
       "7                     yes         NaN  \n",
       "8                     yes         NaN  \n",
       "9                     yes         NaN  \n",
       "10                    yes         NaN  \n",
       "11                    yes         NaN  \n",
       "12                    yes         NaN  \n",
       "13                    yes         NaN  \n",
       "14                    yes         NaN  \n",
       "15                    yes         NaN  \n",
       "16                    yes         NaN  \n",
       "17                    yes         NaN  \n",
       "18                    yes         NaN  \n",
       "19                    yes         NaN  \n",
       "20                    yes         NaN  \n",
       "21                    yes         NaN  \n",
       "22                    yes         NaN  \n",
       "23                    yes         NaN  \n",
       "24                    yes         NaN  \n",
       "25                    yes         NaN  \n",
       "26                    yes         NaN  \n",
       "27                    yes         NaN  \n",
       "28                    yes         NaN  \n",
       "29                    yes         NaN  \n",
       "..                    ...         ...  \n",
       "407                    no         NaN  \n",
       "408                    no         NaN  \n",
       "409                    no         NaN  \n",
       "410                    no         NaN  \n",
       "411                    no         NaN  \n",
       "412                    no         NaN  \n",
       "413                    no         NaN  \n",
       "414                    no         NaN  \n",
       "415                    no         NaN  \n",
       "416                    no         NaN  \n",
       "417                    no         NaN  \n",
       "418                    no         NaN  \n",
       "419                    no         NaN  \n",
       "420                    no         NaN  \n",
       "421                    no         NaN  \n",
       "422                    no         NaN  \n",
       "423                    no         NaN  \n",
       "424                    no         NaN  \n",
       "425                    no         NaN  \n",
       "426                    no         NaN  \n",
       "427                    no         NaN  \n",
       "428                    no         NaN  \n",
       "429                    no         NaN  \n",
       "430                    no         NaN  \n",
       "431                    no         NaN  \n",
       "432                    no         NaN  \n",
       "433                    no         NaN  \n",
       "434                    no         NaN  \n",
       "435                    no         NaN  \n",
       "436                   NaN         NaN  \n",
       "\n",
       "[437 rows x 8 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "permission_new = pd.read_csv('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/sequences_and_permissions_03212018.txt',delimiter='\\t')\n",
    "permission_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "324\n"
     ]
    }
   ],
   "source": [
    "strains_with_permission = []\n",
    "\n",
    "for i in range(len(permission_new)):\n",
    "    record = permission_new.iloc[i]\n",
    "    if record['permission_to_use'] != 'permission_not_received' and record['preliminarily_include'] == 'yes':\n",
    "        strains_with_permission.append(record['strain_name'])\n",
    "\n",
    "print len(strains_with_permission)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "324\n"
     ]
    }
   ],
   "source": [
    "useable_genomes_dict_faunaHeader = {}\n",
    "useable_genomes_dict_augurHeader = {}\n",
    "#making a dict for each kind of header here because it's really nice to have both when troubleshooting.\n",
    "for strain in strains_with_permission:\n",
    "    for key in geoPruned_fauna_dict.keys():\n",
    "        if key.startswith(strain):\n",
    "            useable_genomes_dict_faunaHeader[key] = geoPruned_fauna_dict[key]\n",
    "            useable_genomes_dict_augurHeader[key.split('|')[0]] = geoPruned_fauna_dict[key]\n",
    "\n",
    "print len(useable_genomes_dict_faunaHeader)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10769\n"
     ]
    }
   ],
   "source": [
    "print len(useable_genomes_dict_augurHeader[reference_strain_name])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step 4: Now that we have our set of genomes that we can use, it's time to do some preliminary alignment quality control. I outline genome exclusion criteria in the README of the `data/fastas` directory. Genomes that should be excluded from my analysis are hardcoded in to Augur as `dropped_strains` in `zika.prepare.py`. Most of the process of determining outlier clades is done iteratively based on Augur builds of the tree, however filtering out strains with less that 5000 informative sites is something that I do in this notebook (below)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#write out files that Augur will take in\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/augur-publishedAmericanSeqs-and-ref-03302018.fasta','w') as file:\n",
    "    for key in useable_genomes_dict_faunaHeader.keys():\n",
    "        file.write(str('>' + key + '\\n' + useable_genomes_dict_faunaHeader[key] + '\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22\n",
      "['DOM/2016/MA_WGS16_009', 'DOM/2016/MA_WGS16_020', 'HND/2016/Honduras_1138', 'HND/2016/HU_ME137', 'MEX/2016/mex27', 'BRA/2016/FC_DQ60D1', 'GTM/2016/Guatemala_3294', 'Brazil/2016/ZBRY12', 'BRA/2016/FC_DQ75D1', 'DOM/2016/BB_0085', 'USA/2016/FL019', 'BRA/2016/FC_DQ12D1', 'USA/2016/FL035', 'MEX/2016/mex31', 'MEX/2016/mex50', 'Bahia05', 'COL/FH01/2016', 'Bahia04', 'DOM/2016/BB_0428', 'Bahia15', 'DOM/2016/MA_WGS16_031', 'MEX/2016/Mexico_4227']\n"
     ]
    }
   ],
   "source": [
    "informative_bases = ['A','G','C','T']\n",
    "\n",
    "fraction_informative_bases = {}\n",
    "\n",
    "for key in fauna_dict.keys():\n",
    "    inform_base_count = 0\n",
    "    for base in fauna_dict[key].upper():\n",
    "        if base in informative_bases:\n",
    "            inform_base_count +=1\n",
    "        else:\n",
    "            continue\n",
    "    fraction_informative_bases[key.split('|')[0]] = float(inform_base_count)/len(fauna_dict[key])\n",
    "\n",
    "poor_qual_genomes = {key:value for key,value in fraction_informative_bases.items() if value < 0.5}\n",
    "\n",
    "print len(poor_qual_genomes)\n",
    "print poor_qual_genomes.keys()\n",
    "#wanting to get a sense of how many genomes we could consider poor quality (>50% N's)\n",
    "#we may want to exclude these from the analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "... Interlude here to do all the exploration in Augur ...\n",
    "\n",
    "STEPS: \n",
    "\n",
    "* `cd` in to augur/builds/zika\n",
    "* run `zika_prepare.py` with subsampling off, and pulling from the specific fasta that we wrote out earlier that excludes all non-American isolates except for the french polynesian outgroup. `python zika.prepare.py -v 0 --sequences /Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/augur-publishedAmericanSeqs-and-ref-03302018.fasta`\n",
    "* look at which sequences got excluded (dropped strains) in the `prepared` directory.\n",
    "* iterate through excluding any additional strains by adding them to the `dropped_strains` list in `zika.prepare.py`.\n",
    "\n",
    "Prepare clock filter dropped the following strains:\n",
    "* Bahia05\n",
    "* PF13/251013_18 ... this is the outgroup. What's going on?\n",
    "* Brazil/2016/ZBRC18\n",
    "* DOM/2016/MA_WGS16_020\n",
    "* USA/2016/FLUR001\n",
    "* USA/2016/FLUR057\n",
    "* BRA/2016/FC_DQ12D1\n",
    "\n",
    "Done :)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Step 5: Now that I have the alignment that I want to use, including a good sequence set that is aligned and stripped to reference, I want to make fasta files that I can read in to BEAST that make use of sequences from the multiple sequence aligment, but have fully informative headers (which I'll need to grab from Fauna). \n",
    "\n",
    "Basically, I'll need to use these three dictionaries to do this:\n",
    "\n",
    "* `strain_header_dict` maps strain name in augur form (key) to full fauna fauna header (value).\n",
    "\n",
    "* `msa_dict` maps strain name in augur form (key) to the aligned and stripped-to-reference sequence (value). All sequences in this dictionary should therefore be 10769 nucleotides long, and are in frame."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "320 sequences pulled in from multiple sequence alignment.\n"
     ]
    }
   ],
   "source": [
    "#make strain_header_dict\n",
    "with open(fauna_file,'rU') as file:\n",
    "    strain_header_dict={line.split('|')[0].replace('>',''):line.strip() for line in file if line.startswith('>')}\n",
    "\n",
    "#make msa dict\n",
    "zika_msa = AlignIO.read(open(zika_msa_stripped),'fasta')\n",
    "zika_msa_dict = {record.id:record.seq for record in zika_msa}\n",
    "\n",
    "print '{} sequences pulled in from multiple sequence alignment.'.format(len(zika_msa_dict))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "320\n",
      "['USVI/15/2016', 'USVI/35/2016', 'USVI/45/2016', 'USVI/46/2016']\n"
     ]
    }
   ],
   "source": [
    "output_dict = {}\n",
    "non_fauna_seqs = []\n",
    "for key in zika_msa_dict.keys():\n",
    "    assert len(zika_msa_dict[key]) == 10769\n",
    "    try: \n",
    "        strain_header_dict[key]\n",
    "    except KeyError:\n",
    "        non_fauna_seqs.append(key)\n",
    "        header = key #grab full fauna header from strain_header_dict\n",
    "        seq = zika_msa_dict[key] #get sequence from msa\n",
    "    else:\n",
    "        header = strain_header_dict[key] #grab full fauna header from strain_header_dict\n",
    "        seq = zika_msa_dict[key] #get sequence from msa\n",
    "    output_dict[header] = seq\n",
    "#ensure that everything was trimmed to reference\n",
    "    \n",
    "print len(output_dict)\n",
    "print non_fauna_seqs\n",
    "#print zika_msa_dict['USVI/15/2016']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [],
   "source": [
    "## MAKE BDMM dataset (sample dates need to be in same window as USVI sampling 2016 to 2017)\n",
    "## Have all Brazil, downsample USA, Mexico not needed.\n",
    "\n",
    "from Bio import SeqIO\n",
    "analysis_seq_dict  = SeqIO.to_dict(SeqIO.parse(\"/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/beast-americasZika-04012018.fasta\", \"fasta\"))\n",
    "#print analysis_seq_dict.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Brazil/2016/ZBRX100|2016-05-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_6863|2016-04-12|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BeH819015|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY4|2016-01-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY14|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil_ZKV2015|2015-11-30|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'Brazil/2016/ZBRX2|2016-04-18|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ62D2|2016-04-06|south_america|brazil': Seq('NNNNNNNNNNGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BR_SJRP1840|2016-01-11|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Bahia08|2015-07-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ42D1|2016-03-23|south_america|brazil': Seq('NNNNNNNNNNNNNNNNGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ62D1|2016-03-30|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'SPH2015|2015-03-XX|south_america|brazil': Seq('GACTCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia01|2015-05-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRC16|2016-01-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY7|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY10|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTATCAACA...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_6706|2016-04-08|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRA105|2015-02-23|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ47D1|2016-03-28|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNTTTGAAGCGNAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX7|2016-04-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ68D1|2016-04-01|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ28D1|2016-03-21|south_america|brazil': Seq('NAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRD116|2015-08-28|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRC14|2016-01-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRC28|2016-01-18|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRC321|2015-08-09|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX14|2016-04-24|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ131D1|2016-04-18|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'Brazil/2016/ZBRX130|2016-03-22|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ192D1|2016-05-10|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY8|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY6|2016-02-16|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTATCAACA...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_6696|2016-04-05|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia15|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_6418|2016-03-27|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Rio_U1|2016-01-14|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'BRA/2016/FC_DQ116D1|2016-04-14|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX1|2016-04-18|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX11|2016-04-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_6703|2016-04-08|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia02|2015-05-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'SSABR1|2015-07-01|south_america|brazil': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_5790|2016-03-22|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRE69|2016-04-16|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX15|2016-04-24|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRD107|2015-09-09|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX106|2016-03-07|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX12|2016-04-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX128|2016-03-13|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX13|2016-04-24|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Paraiba_01|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'BRA/2016/FC_DQ75D1|2016-04-04|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX16|2016-04-25|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRC301|2015-05-13|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX6|2016-04-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Natal_RGN|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'BRA/2016/FC_DQ107D1|2016-04-13|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ49D1|2016-03-28|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia05|2015-12-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRD103|2015-08-20|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BeH818995|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia03|2015-05-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRC319|2016-07-10|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BeH819966|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX137|2016-03-03|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX8|2016-04-19|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_6864|2016-04-12|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'HS_2015_BA_01|2015-08-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ12D1|2016-03-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRC313|2015-06-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ122D1|2016-04-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia07|2015-07-01|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BR_SJRP21|2016-02-29|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BeH815744|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX103|2016-05-24|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRC302|2015-05-13|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRC25|2016-01-18|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ5D1|2016-03-14|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Bahia09|2015-08-01|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_DQ58D1|2016-03-30|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY1|2016-01-15|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/PE243/2015|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'BR_SJRP61|2016-04-05|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2015/ZBRC303|2015-05-14|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY12|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BeH828305|2015-XX-XX|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Rio_S1|2016-01-29|south_america|brazil': Seq('GAATCAGACTGCGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...CGG', SingleLetterAlphabet()), 'Brazil/2016/ZBRC18|2016-01-06|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX4|2016-04-18|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX127|2016-03-10|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'BeH823339|2015-XX-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'BRA/2016/FC_5905|2016-03-23|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRY11|2016-01-XX|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), 'Brazil/2016/ZBRX102|2016-02-25|south_america|brazil': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet())}\n",
      "jamaica: has 4 sequences.\n",
      "honduras: has 23 sequences.\n",
      "mexico: has 10 sequences.\n",
      "french_guiana: has 1 sequences.\n",
      "colombia: has 37 sequences.\n",
      "nicaragua: has 8 sequences.\n",
      "brazil: has 95 sequences.\n",
      "usa: has 40 sequences.\n",
      "panama: has 4 sequences.\n",
      "peru: has 2 sequences.\n",
      "french_polynesia: has 1 sequences.\n",
      "venezuela: has 5 sequences.\n",
      "ecuador: has 2 sequences.\n",
      "dominican_republic: has 31 sequences.\n",
      "cuba: has 2 sequences.\n",
      "guatemala: has 3 sequences.\n",
      "usvi: has 34 sequences.\n",
      "martinique: has 3 sequences.\n",
      "haiti: has 4 sequences.\n",
      "suriname: has 2 sequences.\n",
      "el_salvador: has 1 sequences.\n",
      "puerto_rico: has 6 sequences.\n",
      "guadeloupe: has 1 sequences.\n",
      "saint_barthelemy: has 1 sequences.\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "country_binning_dict = defaultdict(dict)\n",
    "\n",
    "for key in analysis_seq_dict.keys():\n",
    "    country = key.split('|')[3]\n",
    "    country_binning_dict[country].update({key:analysis_seq_dict[key].seq})\n",
    "    \n",
    "#print country_binning_dict['brazil']\n",
    "for key in country_binning_dict.keys():\n",
    "    print \"{0}: has {1} sequences.\".format(key, len(country_binning_dict[key]))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "brazil: has 59 sequences.\n",
      "jamaica: has 4 sequences.\n",
      "suriname: has 1 sequences.\n",
      "honduras: has 22 sequences.\n",
      "dominican_republic: has 31 sequences.\n",
      "usa: has 40 sequences.\n",
      "mexico: has 7 sequences.\n",
      "nicaragua: has 8 sequences.\n",
      "guatemala: has 1 sequences.\n",
      "el_salvador: has 1 sequences.\n",
      "puerto_rico: has 5 sequences.\n",
      "guadeloupe: has 1 sequences.\n",
      "colombia: has 10 sequences.\n",
      "peru: has 2 sequences.\n",
      "martinique: has 2 sequences.\n",
      "haiti: has 3 sequences.\n",
      "usvi: has 34 sequences.\n",
      "venezuela: has 5 sequences.\n",
      "saint_barthelemy: has 1 sequences.\n"
     ]
    }
   ],
   "source": [
    "country_binned_dict2016 = defaultdict(dict)\n",
    "\n",
    "for key in analysis_seq_dict.keys():\n",
    "    country = key.split('|')[3]\n",
    "    date = key.split('|')[1]\n",
    "    if date.startswith('2016') and 'X' not in date:\n",
    "        country_binned_dict2016[country].update({key:analysis_seq_dict[key].seq})\n",
    "    else:\n",
    "        continue\n",
    "\n",
    "for key in country_binned_dict2016.keys():\n",
    "    print \"{0}: has {1} sequences.\".format(key, len(country_binned_dict2016[key]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### BDMM genomes subset - criteria:\n",
    "\n",
    "* All samples need to follow similar time window to USVI (ie. sampling date in 2016).\n",
    "* Have full sampling date known.\n",
    "* Want to downsample USA (outbreak density is going to confuse BD rates and multi-type designation).\n",
    "* Remove Colombian highly sequenced Barranquilla outbreak.\n",
    "* See no interdigitation of USVI samples with samples from: Honduras, Mexico \n",
    "\n",
    "####  POSSIBLE DATASETS\n",
    "* What about just a USVI-Brazil dataset?\n",
    "* USVI complete, but everything else subsampled down?\n",
    "* USVI + Caribbean + Brazil + Venezuela + Suriname + Puerto Rico + Subsampled USA + Dominican Republic (close interdigitating countries).\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def is_sampling_date_known(taxon_name, date_index_position, delimiter = '|'):\n",
    "    \"\"\" returns bool True if year, month, and day are known for a sample given the taxon name, else False \"\"\"\n",
    "    flag = True\n",
    "    date_string = taxon_name.split(delimiter)[date_index_position].upper()\n",
    "    if 'X' in date_string:\n",
    "        flag = False\n",
    "    return flag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Martinique/2016/FL001|2016-03-22|north_america|martinique': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNACAGTATCAACA...NNN', SingleLetterAlphabet()), 'Martinique/2016/FL001Sa|2016-03-22|north_america|martinique': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())}\n"
     ]
    }
   ],
   "source": [
    "print country_binned_dict2016['martinique']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14\n"
     ]
    }
   ],
   "source": [
    "#print country_seq_dict['brazil'].items()\n",
    "import random\n",
    "countries_to_exclude = ['honduras', 'mexico', 'el_salvador', 'nicaragua', 'guatemala' ,'panama']\n",
    "countries_to_subsample = ['usa', 'dominican_republic', 'brazil']\n",
    "\n",
    "bdmm_alignment = {}\n",
    "\n",
    "for key in country_binned_dict2016.keys():\n",
    "    if key in countries_to_exclude:\n",
    "        continue\n",
    "    elif key in countries_to_subsample:\n",
    "        sample = random.sample(country_binned_dict2016[key].items(), 10) #produces a list of tuples of (name,sequence) that were randomly sampled\n",
    "        bdmm_alignment[key] = sample\n",
    "    else:\n",
    "        bdmm_alignment[key] = [(strain,country_binned_dict2016[key][strain]) for strain in country_binned_dict2016[key].keys()]\n",
    "        \n",
    "print len(bdmm_alignment)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime as dt\n",
    "\n",
    "def decimalDate(date,date_format=\"%Y-%m-%d\"):\n",
    "    \"\"\" Converts calendar dates in specified format to decimal date. \"\"\"\n",
    "    sampling_date = dt.strptime(date,date_format)\n",
    "    year =  sampling_date.year\n",
    "    beginning_of_year = datetime.datetime(year,1,1) #year, month, day, so year 1,1 == beginning of year\n",
    "    end_of_year = datetime.datetime(year+1,1,1) #next year 1,1 is a full year away from beginning of sampling year\n",
    "    #return fraction of the full year (in seconds) that occurs between jan 1st and your sampling date\n",
    "    return year + ((sampling_date - beginning_of_year).total_seconds() / ((end_of_year - beginning_of_year).total_seconds()))\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/bdmm_alignment_04052018.fasta','w') as file:\n",
    "    for key in bdmm_alignment.keys():\n",
    "        if key == 'usvi':\n",
    "            deme = 'usvi'\n",
    "        else:\n",
    "            deme = 'nonusvi'\n",
    "        for i in range(len(bdmm_alignment[key])): #have a list of tuples (taxon,sequence)\n",
    "            header = bdmm_alignment[key][i][0]#first element in the tuple ins the taxon name\n",
    "            date = decimalDate(header.split('|')[1])\n",
    "            assert len(bdmm_alignment[key][i][1]) == 10769\n",
    "            file.write(str('>'+ header.split('|')[0] +'|'+ str(date) + '|'+ header.split('|')[3] + '|' +deme+ '\\n' + str(bdmm_alignment[key][i][1]) + '\\n'))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USVI/35/2016\n",
      "USVI/15/2016\n",
      "USVI/45/2016\n",
      "USVI/46/2016\n"
     ]
    }
   ],
   "source": [
    "#write output dict to fasta\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/beast-americasZika-04012018.fasta','w') as out_file:\n",
    "    for key in output_dict.keys():\n",
    "        \n",
    "        if key in non_fauna_seqs:\n",
    "            print key\n",
    "            out_file.write(str('>' + key + '\\n' + output_dict[key] + '\\n'))\n",
    "        else:\n",
    "            split_name = key.split('|')\n",
    "            header = split_name[0] +'|'+ split_name[3] + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "            out_file.write(str(header + '\\n' + output_dict[key] + '\\n'))\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'output_dict' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-103-51539e3c8e0b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi_primary_clade.fasta'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mout_file\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;32min\u001b[0m \u001b[0moutput_dict\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'|'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'>'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m''\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32min\u001b[0m \u001b[0musvi_primary_clade\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m             \u001b[0msplit_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'|'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'output_dict' is not defined"
     ]
    }
   ],
   "source": [
    "#as inferred with subtree parsing in Baltic\n",
    "usvi_primary_clade = ['USVI/3/2016', 'USVI/5/2016', 'USVI/34/2016', 'USVI/13/2016', 'USVI/38/2016', 'USVI/43/2016', \n",
    "                      'USVI/6/2016', 'USVI/21/2016', 'USVI/25/2016', 'USVI/27/2016', 'USIV/35/2016', 'USVI/2/2016',\n",
    "                      'USVI/1/2016', 'USVI/37/2016', 'USVI/12/2016', 'USVI/39/2016', 'USVI/30/2016', 'USVI/32/2016', \n",
    "                      'USVI/23/2016']\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi_primary_clade.fasta','w') as out_file:\n",
    "    for key in output_dict.keys():\n",
    "        if key.split('|')[0].replace('>','') in usvi_primary_clade:\n",
    "            split_name = key.split('|')\n",
    "            header = split_name[0] +'|'+ split_name[3] + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "            out_file.write(str(header + '\\n' + output_dict[key] + '\\n'))\n",
    "            \n",
    "usvi_secondary_clade = ['USVI/41/2016', 'USVI/19/2016', 'USVI/20/2016', 'USVI/42/2016', 'USVI/28/2016']\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi_secondary_clade.fasta','w') as out_file:\n",
    "    for key in output_dict.keys():\n",
    "        if key.split('|')[0].replace('>','') in usvi_secondary_clade:\n",
    "            split_name = key.split('|')\n",
    "            header = split_name[0] +'|'+ split_name[3] + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "            out_file.write(str(header + '\\n' + output_dict[key] + '\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BRA/2016/FC_DQ107D1|2016.28142077|south_america|brazil\n",
      ">BRA/2016/FC_DQ62D1|2016.2431694|south_america|brazil\n",
      ">Colombia/2016/ZC204Se|2016.0136612|south_america|colombia\n",
      ">Z16019|2016.15300546|south_america|venezuela\n",
      ">BRA/2016/FC_DQ5D1|2016.19945355|south_america|brazil\n",
      ">Brazil/2016/ZBRC319|2016.52185792|south_america|brazil\n",
      ">GZ02/2016|2016.15027322|south_america|venezuela\n",
      ">BRA/2016/FC_DQ28D1|2016.21857923|south_america|brazil\n",
      ">USVI/36/2016|2016.69945355|north_america|usvi\n",
      ">USA/2016/FLUR008|2016.59016393|north_america|usa\n",
      ">Nica2_16|2016.03278689|north_america|nicaragua\n",
      ">USA/2016/FLUR022|2016.66393443|north_america|usa\n",
      ">PAN/CDC_259359_V1_V3/2015|2015.96164384|north_america|panama\n",
      ">GZ01|2016.12021858|south_america|venezuela\n",
      ">USVI/38/2016|2016.81420765|north_america|usvi\n",
      ">HND/2016/HU_ME167|2016.43715847|north_america|honduras\n",
      ">DOM/2016/BB_0180|2016.29508197|north_america|dominican_republic\n",
      ">FHT1175/HON/2016|2016.65027322|north_america|honduras\n",
      ">Brazil/2016/ZBRX6|2016.29781421|south_america|brazil\n",
      ">Brazil/2016/ZBRX130|2016.22131148|south_america|brazil\n",
      ">DOM/2016/BB_0436|2016.45081967|north_america|dominican_republic\n",
      ">V103344|2015.91506849|north_america|guatemala\n",
      ">Brazil/2016/ZBRC18|2016.0136612|south_america|brazil\n",
      ">Brazil/2016/ZBRY7|2016.0|south_america|brazil\n",
      ">Colombia/2016/ZC199Se|2016.02459016|south_america|colombia\n",
      ">SL1602|2016.05737705|south_america|suriname\n",
      ">USVI/41/2016|2016.8579235|north_america|usvi\n",
      ">USA/2016/FL030|2016.58469945|north_america|usa\n",
      ">HN16|2016.0|north_america|honduras\n",
      ">DOM/2016/BB_0091|2016.26502732|north_america|dominican_republic\n",
      ">USA/2016/FLUR005|2016.57377049|north_america|usa\n",
      ">DOM/2016/MA_WGS16_007|2016.35519126|north_america|dominican_republic\n",
      ">COL/2016/SU_2293A|2016.43715847|south_america|colombia\n",
      ">FPI15263/PERU/Loreto/2016|2016.52459016|south_america|peru\n",
      ">Brazil/2015/ZBRC303|2015.36438356|south_america|brazil\n",
      ">COL/UF_1/2016|2016.10655738|south_america|colombia\n",
      ">COL/2016/SU_1856A|2016.32240437|south_america|colombia\n",
      ">DOM/2016/BB_0269|2016.31967213|north_america|dominican_republic\n",
      ">USVI/20/2016|2016.78142077|north_america|usvi\n",
      ">MRS_OPY_Martinique_PaRi_2015|2015.0|north_america|martinique\n",
      ">UNK/2016/MA_WGS16_029|2016.59289617|north_america|el_salvador\n",
      ">PRI/2016/MA_WGS16_005|2016.27868852|north_america|puerto_rico\n",
      ">BRA/2016/FC_DQ122D1|2016.28688525|south_america|brazil\n",
      ">Brazil/2015/ZBRC302|2015.36164384|south_america|brazil\n",
      ">USA/2016/FLUR026|2016.69945355|north_america|usa\n",
      ">COL/2016/SU_2724A|2016.25956284|south_america|colombia\n",
      ">PRVABC59|2015.0|north_america|puerto_rico\n",
      ">USA/2016/FL032|2016.59289617|north_america|usa\n",
      ">DOM/2016/BB_0115|2016.27595628|north_america|dominican_republic\n",
      ">Brazil/2016/ZBRX11|2016.29781421|south_america|brazil\n",
      ">NIC/4886_12A1/2016|2016.52185792|north_america|nicaragua\n",
      ">Haiti/1225/2014|2014.94520548|north_america|haiti\n",
      ">USA/2016/FLUR013|2016.6420765|north_america|usa\n",
      ">Brazil/2016/ZBRX127|2016.18852459|south_america|brazil\n",
      ">USA/2016/FLUR015|2016.64480874|north_america|usa\n",
      ">HND/2016/HU_ME152|2016.42896175|north_america|honduras\n",
      ">USVI/42/2016|2016.81693989|north_america|usvi\n",
      ">DOM/2016/BB_0085|2016.26502732|north_america|dominican_republic\n",
      ">R116265|2016.47540984|north_america|mexico\n",
      ">HND/R103451/2015|2015.01369863|north_america|honduras\n",
      ">Aedes_aegypti/USA/2016/FL06|2016.71857923|north_america|usa\n",
      ">PF13/251013_18|2013.81369863|oceania|french_polynesia\n",
      ">Brazil_ZKV2015|2015.91232877|south_america|brazil\n",
      ">NIC/8610_13A1/2016|2016.5136612|north_america|nicaragua\n",
      ">USVI/7/2016|2016.81967213|north_america|usvi\n",
      ">USVI/19/2016|2016.88797814|north_america|usvi\n",
      ">USVI/11/2016|2016.22131148|north_america|usvi\n",
      ">HND/2016/HU_ME131|2016.40437158|north_america|honduras\n",
      ">Brazil/2016/ZBRX2|2016.29508197|south_america|brazil\n",
      ">Colombia/2016/ZC188Se|2016.04098361|south_america|colombia\n",
      ">BRA/2016/FC_6703|2016.26775956|south_america|brazil\n",
      ">Aedes_aegypti/USA/2016/FL02|2016.6420765|north_america|usa\n",
      ">Aedes_aegypti/USA/2016/FL04|2016.67486339|north_america|usa\n",
      ">USA/2016/FLUR002|2016.57103825|north_america|usa\n",
      ">HND/2016/HU_ME58|2016.36338798|north_america|honduras\n",
      ">USA/2016/FL029|2016.58196721|north_america|usa\n",
      ">USIV/35/2016|2016.68579235|north_america|usvi\n",
      ">Bahia07|2015.49589041|south_america|brazil\n",
      ">USVI/40/2016|2016.8442623|north_america|usvi\n",
      ">USA/2016/FLUR011|2016.6010929|north_america|usa\n",
      ">USVI/25/2016|2016.73770492|north_america|usvi\n",
      ">USA/2016/FL022|2016.55464481|north_america|usa\n",
      ">BRA/2016/FC_DQ62D2|2016.26229508|south_america|brazil\n",
      ">Brazil/2016/ZBRX7|2016.29781421|south_america|brazil\n",
      ">USVI/37/2016|2016.76229508|north_america|usvi\n",
      ">Brazil/2016/ZBRX106|2016.18032787|south_america|brazil\n",
      ">USA/2016/FLSR036|2016.71584699|north_america|usa\n",
      ">Martinique/2016/FL001Sa|2016.22131148|north_america|martinique\n",
      ">Brazil/2016/ZBRX8|2016.29781421|south_america|brazil\n",
      ">DOM/2016/MA_WGS16_040|2016.76229508|north_america|dominican_republic\n",
      ">Brazil/2016/ZBRY8|2016.0|south_america|brazil\n",
      ">HND/2016/HU_SZ76|2016.33606557|north_america|honduras\n",
      ">DOM/2016/BB_0216|2016.30327869|north_america|dominican_republic\n",
      ">HND/2016/HU_ME33|2016.32240437|north_america|honduras\n",
      ">Brazil/2016/ZBRX12|2016.29781421|south_america|brazil\n",
      ">PRI/2016/MA_WGS16_004|2016.28142077|north_america|puerto_rico\n",
      ">HS_2015_BA_01|2015.0|south_america|brazil\n",
      ">EcEs089_16|2016.0|south_america|ecuador\n",
      ">USVI/32/2016|2016.60928962|north_america|usvi\n",
      ">USA/2016/FLUR063|2016.75409836|north_america|usa\n",
      ">VEN/UF_1/2016|2016.2295082|south_america|venezuela\n",
      ">Haiti/1/2016|2016.09562842|north_america|haiti\n",
      ">NIC/7252_12A1/2016|2016.49180328|north_america|nicaragua\n",
      ">PuertoRico/2016/FL008U|2016.46994536|north_america|puerto_rico\n",
      ">FB_GWUH_2016|2016.08743169|north_america|guatemala\n",
      ">SPH2015|2015.0|south_america|brazil\n",
      ">FVM00318/VEN/Maracay/2016|2016.79781421|south_america|venezuela\n",
      ">USVI/34/2016|2016.58196721|north_america|usvi\n",
      ">PuertoRico/2016/FL016U|2016.50273224|north_america|puerto_rico\n",
      ">BRA/2016/FC_6863|2016.27868852|south_america|brazil\n",
      ">Rio_U1|2016.03551913|south_america|brazil\n",
      ">Brazil/2016/ZBRX103|2016.39344262|south_america|brazil\n",
      ">DOM/2016/MA_WGS16_013|2016.45355191|north_america|dominican_republic\n",
      ">Aedessp/MEX/MEX_I_7/2016|2016.0|north_america|mexico\n",
      ">HND/2016/HU_ME42|2016.35245902|north_america|honduras\n",
      ">Brazil/PE243/2015|2015.0|south_america|brazil\n",
      ">Nica1_16|2016.04918033|north_america|nicaragua\n",
      ">JAM/2016/MA_WGS16_025|2016.52185792|north_america|jamaica\n",
      ">SSABR1|2015.49589041|south_america|brazil\n",
      ">USVI/1/2016|2016.74043716|north_america|usvi\n",
      ">V8375|2015.83287671|north_america|guatemala\n",
      ">HND/2016/HU_ME59|2016.36338798|north_america|honduras\n",
      ">Brazil/2015/ZBRD116|2015.65479452|south_america|brazil\n",
      ">Brazil/2015/ZBRD103|2015.63287671|south_america|brazil\n",
      ">JAM/2016/MA_WGS16_041|2016.77868852|north_america|jamaica\n",
      ">PAN/CDC_259364_V1_V2/2015|2015.96164384|north_america|panama\n",
      ">Brazil/2016/ZBRC25|2016.04644809|south_america|brazil\n",
      ">Brazil/2016/ZBRE69|2016.28961749|south_america|brazil\n",
      ">DOM/2016/BB_0071|2016.25956284|north_america|dominican_republic\n",
      ">USVI/30/2016|2016.59836066|north_america|usvi\n",
      ">Brazil/2016/ZBRX16|2016.31420765|south_america|brazil\n",
      ">BRA/2016/FC_DQ192D1|2016.35519126|south_america|brazil\n",
      ">JAM/2016/MA_WGS16_039|2016.74043716|north_america|jamaica\n",
      ">BRA/2016/FC_6706|2016.26775956|south_america|brazil\n",
      ">USVI/23/2016|2016.5273224|north_america|usvi\n",
      ">COL/2016/SU_1810A|2016.26229508|south_america|colombia\n",
      ">PHE_Guadeloupe|2016.30327869|north_america|guadeloupe\n",
      ">USVI/3/2016|2016.73497268|north_america|usvi\n",
      ">NIC/6406_13A1/2016|2016.40710383|north_america|nicaragua\n",
      ">FHT1166/HON/2016|2016.65027322|north_america|honduras\n",
      ">DOM/2016/BB_0208|2016.30054645|north_america|dominican_republic\n",
      ">Bahia08|2015.53424658|south_america|brazil\n",
      ">Brazil/2016/ZBRC16|2016.04918033|south_america|brazil\n",
      ">MEX_CIENI551|2016.16939891|north_america|mexico\n",
      ">Brazil/2016/ZBRY6|2016.12568306|south_america|brazil\n",
      ">HND/2016/HU_ME136|2016.41256831|north_america|honduras\n",
      ">Brazil/2016/ZBRX1|2016.29508197|south_america|brazil\n",
      ">BRA/2016/FC_6696|2016.25956284|south_america|brazil\n",
      ">Brazil/2016/ZBRX137|2016.16939891|south_america|brazil\n",
      ">USVI/6/2016|2016.79781421|north_america|usvi\n",
      ">Aedes_aegypti/USA/2016/FL01|2016.63934426|north_america|usa\n",
      ">Brazil/2015/ZBRD107|2015.68767123|south_america|brazil\n",
      ">Brazil/2015/ZBRC321|2015.60273973|south_america|brazil\n",
      ">USA/2016/FLUR007|2016.59016393|north_america|usa\n",
      ">Brazil/2016/ZBRY4|2016.03825137|south_america|brazil\n",
      ">EcEs062_16|2016.0|south_america|ecuador\n",
      ">Aedessp/MEX/MEX_2_81/2016|2016.0|north_america|mexico\n",
      ">DOM/2016/BB_0127|2016.27595628|north_america|dominican_republic\n",
      ">PAN/BEI_259634_V4/2016|2016.0|north_america|panama\n",
      ">USVI/28/2016|2016.90710383|north_america|usvi\n",
      ">Brazil/2016/ZBRX4|2016.29508197|south_america|brazil\n",
      ">HND/2016/HU_ME147|2016.42349727|north_america|honduras\n",
      ">USVI/39/2016|2016.85519126|north_america|usvi\n",
      ">Brazil/2016/ZBRY10|2016.0|south_america|brazil\n",
      ">MEX_I_7|2015.0|north_america|mexico\n",
      ">USA/2016/FLUR058|2016.77595628|north_america|usa\n",
      ">R114916|2016.42896175|north_america|dominican_republic\n",
      ">HND/2016/HU_ME180|2016.44808743|north_america|honduras\n",
      ">Brazil/2015/ZBRC301|2015.36164384|south_america|brazil\n",
      ">Cuba_2017|2017.11506849|north_america|cuba\n",
      ">USA/2016/FLUR006|2016.57377049|north_america|usa\n",
      ">BRA/2016/FC_DQ49D1|2016.23770492|south_america|brazil\n",
      ">Brazil/2016/ZBRY11|2016.0|south_america|brazil\n",
      ">V103451|2016.0136612|north_america|honduras\n",
      ">Bahia02|2015.0|south_america|brazil\n",
      ">USA/2016/FLWB044|2016.74043716|north_america|usa\n",
      ">Natal_RGN|2015.0|south_america|brazil\n",
      ">Aedes_aegypti/USA/2016/FL03|2016.6420765|north_america|usa\n",
      ">USVI/13/2016|2016.6147541|north_america|usvi\n",
      ">Brazil/2016/ZBRX15|2016.31147541|south_america|brazil\n",
      ">Brazil/2016/ZBRX13|2016.31147541|south_america|brazil\n",
      ">Aedes_aegypti/MEX/MEX_I_44/2016|2016.0|north_america|mexico\n",
      ">Bahia03|2015.0|south_america|brazil\n",
      ">BRA/2016/FC_6418|2016.23497268|south_america|brazil\n",
      ">USVI/4/2016|2016.78415301|north_america|usvi\n",
      ">DOM/2016/BB_0433|2016.44808743|north_america|dominican_republic\n",
      ">USA/2016/Hu0015SA|2016.64480874|north_america|usa\n",
      ">Brazil/2016/ZBRX14|2016.31147541|south_america|brazil\n",
      ">DOM/2016/MA_WGS16_036|2016.72404372|north_america|dominican_republic\n",
      ">USA/2016/FLWB042|2016.73497268|north_america|usa\n",
      ">USA/2016/FLSR043|2016.74043716|north_america|usa\n",
      ">Rio_S1|2016.07650273|south_america|brazil\n",
      ">USA/2016/FL028|2016.58196721|north_america|usa\n",
      ">Dominican_Republic/2016/PD1|2016.08469945|north_america|dominican_republic\n",
      ">V17271|2015.0|south_america|french_guiana\n",
      ">USVI/12/2016|2016.84153005|north_america|usvi\n",
      ">BRA/2016/FC_DQ131D1|2016.29508197|south_america|brazil\n",
      ">Bahia01|2015.0|south_america|brazil\n",
      ">USA/2016/FLUR001|2016.57103825|north_america|usa\n",
      ">NIC/5005_13A1/2016|2016.30601093|north_america|nicaragua\n",
      ">Z1106033|2015.75068493|south_america|suriname\n",
      ">MEX/InDRE/2016|2016.15027322|north_america|mexico\n",
      ">HND/2016/HU_ME38|2016.32786885|north_america|honduras\n",
      ">HND/2016/HU_ME156|2016.43169399|north_america|honduras\n",
      ">USA/UT_1/2016|2016.0|north_america|mexico\n",
      ">USA/2016/FL039|2016.62568306|north_america|usa\n",
      ">Brazil/2016/ZBRX100|2016.37978142|south_america|brazil\n",
      ">Aedes_aegypti/USA/2016/FL05|2016.68852459|north_america|usa\n",
      ">DOM/2016/MA_WGS16_024|2016.5136612|north_america|dominican_republic\n",
      ">USVI/21/2016|2016.84972678|north_america|usvi\n",
      ">BRA/2016/FC_5790|2016.22131148|south_america|brazil\n",
      ">USVI/5/2016|2016.79234973|north_america|usvi\n",
      ">Aedes_aegypti/USA/2016/FL08|2016.75956284|north_america|usa\n",
      ">USA/2016/FL021|2016.54644809|north_america|usa\n",
      ">HND/2016/HU_ME171|2016.43989071|north_america|honduras\n",
      ">DOM/2016/MA_WGS16_011|2016.42896175|north_america|dominican_republic\n",
      ">COL/FLR/2015|2015.0|south_america|colombia\n",
      ">FPI15198/PERU/Loreto/2016|2016.48907104|south_america|peru\n",
      ">BRA/2016/FC_DQ58D1|2016.2431694|south_america|brazil\n",
      ">BRA/2016/FC_DQ47D1|2016.23770492|south_america|brazil\n",
      ">USVI/22/2016|2016.68306011|north_america|usvi\n",
      ">HND/2016/HU_ME50|2016.36065574|north_america|honduras\n",
      ">USVI/2/2016|2016.74043716|north_america|usvi\n",
      ">Haiti/2016/PD|2016.08469945|north_america|haiti\n",
      ">Brazil/2016/ZBRY1|2016.03825137|south_america|brazil\n",
      ">Brazil/2016/ZBRC14|2016.03825137|south_america|brazil\n",
      ">Cuba/2017/Hu0046Sa|2017.29589041|north_america|cuba\n",
      ">HND/2016/HU_ME178|2016.43989071|north_america|honduras\n",
      ">PRI/2016/MA_WGS16_016|2016.48360656|north_america|puerto_rico\n",
      ">DOM/2016/BB_0076|2016.25956284|north_america|dominican_republic\n",
      ">USA/2016/MA_WGS16_034|2016.62021858|north_america|usa\n",
      ">BRA/2016/FC_DQ116D1|2016.28415301|south_america|brazil\n",
      ">DOM/2016/BB_0183|2016.29508197|north_america|dominican_republic\n",
      ">Colombia/2016/ZC207Se|2016.02185792|south_america|colombia\n",
      ">USA/2016/FL038|2016.59289617|north_america|usa\n",
      ">BRA/2016/FC_6864|2016.27868852|south_america|brazil\n",
      ">Paraiba_01|2015.0|south_america|brazil\n",
      ">Brazil/2016/ZBRX128|2016.19672131|south_america|brazil\n",
      ">USA/2016/FL010|2016.4726776|north_america|usa\n",
      ">PAN/CDC_259249_V1_V3/2015|2015.94246575|north_america|panama\n",
      ">NIC/6188_13A1/2016|2016.49453552|north_america|nicaragua\n",
      ">Colombia/2016/ZC192Se|2016.01639344|south_america|colombia\n",
      ">BRA/2016/FC_5905|2016.22404372|south_america|brazil\n",
      ">Bahia09|2015.58082192|south_america|brazil\n",
      ">USVI/43/2016|2016.54644809|north_america|usvi\n",
      ">Brazil/2015/ZBRA105|2015.14520548|south_america|brazil\n",
      ">USVI/24/2016|2016.69945355|north_america|usvi\n",
      ">USA/2016/FLUR009|2016.59016393|north_america|usa\n",
      ">DOM/2016/BB_0059|2016.2568306|north_america|dominican_republic\n",
      ">Martinique/2016/FL001|2016.22131148|north_america|martinique\n",
      ">HND/2016/HU_ME172|2016.43989071|north_america|honduras\n",
      ">USVI/27/2016|2016.63114754|north_america|usvi\n",
      ">USA/2016/FL036|2016.59016393|north_america|usa\n",
      ">Brazil/2016/ZBRX102|2016.15027322|south_america|brazil\n",
      ">BRA/2016/FC_DQ42D1|2016.22404372|south_america|brazil\n",
      ">DOM/2016/MA_WGS16_014|2016.46174863|north_america|dominican_republic\n",
      ">BRA/2016/FC_DQ75D1|2016.2568306|south_america|brazil\n"
     ]
    }
   ],
   "source": [
    "from datetime import datetime as dt\n",
    "\n",
    "def decimalDate(date,date_format=\"%Y-%m-%d\"):\n",
    "    \"\"\" Converts calendar dates in specified format to decimal date. \"\"\"\n",
    "    sampling_date = dt.strptime(date,date_format)\n",
    "    year =  sampling_date.year\n",
    "    beginning_of_year = datetime.datetime(year,1,1) #year, month, day, so year 1,1 == beginning of year\n",
    "    end_of_year = datetime.datetime(year+1,1,1) #next year 1,1 is a full year away from beginning of sampling year\n",
    "    #return fraction of the full year (in seconds) that occurs between jan 1st and your sampling date\n",
    "    return year + ((sampling_date - beginning_of_year).total_seconds() / ((end_of_year - beginning_of_year).total_seconds()))\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi_primary_clade_decimalDate.fasta','w') as out_file:\n",
    "    for key in output_dict.keys():\n",
    "        if key.split('|')[0].replace('>','') in usvi_primary_clade:\n",
    "            split_name = key.split('|')\n",
    "            date = decimalDate(split_name[3])\n",
    "            header = split_name[0] +'|'+ str(date) + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "            out_file.write(str(header + '\\n' + output_dict[key] + '\\n'))\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/beast-americasZika-outgrouped_decimalDate.fasta','w') as out_file:\n",
    "    for key in output_dict.keys():\n",
    "            split_name = key.split('|')\n",
    "            calendarDate = split_name[3]\n",
    "            if 'XX' in calendarDate.split('-'):\n",
    "                date = float(calendarDate[:4])\n",
    "            else:\n",
    "                date = decimalDate(calendarDate)\n",
    "            header = split_name[0] +'|'+ str(date) + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "            print header\n",
    "            out_file.write(str(header + '\\n' + output_dict[key] + '\\n'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make fasta files that represent more equitably-subsampled demes\n",
    "\n",
    "There is a large range in how many zika genomes have been sampled from each country. To ensure that this sampling bias isn't driving the phylogeographic dynamics we infer, I'm subsampling the alignment to make the sampling per country more even. Aiming to have 10 genomes per country."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "nested_dict = defaultdict(dict) #nested dict e.g. {brazil: {seq1:ACCT, seq2:ACGGGTT}}\n",
    "for key in output_dict.keys():\n",
    "    split_name = key.split('|')\n",
    "    nested_dict[split_name[5]].update({key:output_dict[key]})\n",
    "\n",
    "country_seq_dict = dict(nested_dict) #avoid continually adding to default dict accidentally."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print country_seq_dict['brazil'].items()\n",
    "import random\n",
    "\n",
    "samp = random.sample(country_seq_dict['brazil'].items(), 10) #produces a list of tuples of (name,sequence) that were randomly sampled\n",
    "#print samp[0][1]\n",
    "#print 'There are {} countries have at least one genome.'.format(len(subsampled_demes.keys()))\n",
    "#print subsampled_demes['french_polynesia']\n",
    "#print subsampled_demes['honduras'][0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['brazil', 'jamaica', 'honduras', 'dominican_republic', 'usa', 'mexico', 'venezuela', 'panama', 'puerto_rico', 'guatemala', 'colombia', 'martinique', 'haiti', 'usvi', 'nicaragua']\n",
      "['brazil', 'jamaica', 'honduras', 'dominican_republic', 'usa', 'mexico', 'venezuela', 'panama', 'puerto_rico', 'guatemala', 'colombia', 'martinique', 'haiti', 'usvi', 'nicaragua']\n",
      "['jamaica', 'honduras', 'dominican_republic', 'mexico', 'french_guiana', 'guatemala', 'colombia', 'martinique', 'haiti', 'nicaragua', 'brazil', 'suriname', 'usa', 'panama', 'cuba', 'el_salvador', 'puerto_rico', 'guadeloupe', 'peru', 'french_polynesia', 'usvi', 'venezuela', 'ecuador']\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print new_glm_set_subsampled.keys()\n",
    "print new_glm_set_all_samples.keys()\n",
    "print country_seq_dict.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_glm_set_all_samples = {}\n",
    "new_glm_set_subsampled = {}\n",
    "new_glm_set_subsampled_except_usvi= {}\n",
    "\n",
    "for country in country_seq_dict.keys():\n",
    "    if len(country_seq_dict[country]) > 2: #to be included in analysis a country needs at least 3 genomes\n",
    "        new_glm_set_all_samples[country] = country_seq_dict[country]\n",
    "            \n",
    "        if country == 'usvi':\n",
    "            new_glm_set_subsampled_except_usvi[country] = [(key,country_seq_dict[key]) for key in country_seq_dict.keys()]\n",
    "        else:     \n",
    "            if len(country_seq_dict[country]) > 10:\n",
    "                sample = random.sample(country_seq_dict[country].items(), 10)\n",
    "                new_glm_set_subsampled[country] = sample\n",
    "                new_glm_set_subsampled_except_usvi[country] = sample\n",
    "                \n",
    "            else:\n",
    "                new_glm_set_subsampled[country] = [(key,country_seq_dict[key]) for key in country_seq_dict.keys()]\n",
    "                new_glm_set_subsampled_except_usvi[country] = [(key,country_seq_dict[key]) for key in country_seq_dict.keys()]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31\n",
      "10\n",
      "{'>USVI/43/2016|zika|VI43|2016-07-19|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/7/2016|zika|VI7|2016-10-27|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/28/2016|zika|VI28|2016-11-28|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/12/2016|zika|VI12|2016-11-04|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/39/2016|zika|VI39|2016-11-09|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/19/2016|zika|VI19|2016-11-21|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/23/2016|zika|VI23|2016-07-12|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/11/2016|zika|VI11|2016-03-22|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/34/2016|zika|VI34|2016-08-01|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/36/2016|zika|VI36|2016-09-13|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/3/2016|zika|VI3|2016-09-26|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet()), '>USIV/35/2016|zika|VI35|2016-09-08|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/40/2016|zika|VI40|2016-11-05|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/25/2016|zika|VI25|2016-09-27|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/2/2016|zika|VI2|2016-09-28|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/1/2016|zika|VI1|2016-09-28|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/13/2016|zika|VI13|2016-08-13|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/37/2016|zika|VI37|2016-10-06|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/24/2016|zika|VI24|2016-09-13|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCGTGAGCGNNNAACAGGATCAACA...NNN', SingleLetterAlphabet()), '>USVI/42/2016|zika|VI42|2016-10-26|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/4/2016|zika|VI4|2016-10-14|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/6/2016|zika|VI6|2016-10-19|north_america|usvi|saint_john|saint_john|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/22/2016|zika|VI22|2016-09-07|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/27/2016|zika|VI27|2016-08-19|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/20/2016|zika|VI20|2016-10-13|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/41/2016|zika|VI41|2016-11-10|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/38/2016|zika|VI38|2016-10-25|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/5/2016|zika|VI5|2016-10-17|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/21/2016|zika|VI21|2016-11-07|north_america|usvi|saint_john|saint_john|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/30/2016|zika|VI30|2016-08-07|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()), '>USVI/32/2016|zika|VI32|2016-08-11|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?': Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())}\n",
      "[('>USVI/6/2016|zika|VI6|2016-10-19|north_america|usvi|saint_john|saint_john|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/24/2016|zika|VI24|2016-09-13|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCGTGAGCGNNNAACAGGATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/32/2016|zika|VI32|2016-08-11|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/5/2016|zika|VI5|2016-10-17|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/11/2016|zika|VI11|2016-03-22|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/25/2016|zika|VI25|2016-09-27|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/36/2016|zika|VI36|2016-09-13|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/3/2016|zika|VI3|2016-09-26|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNN', SingleLetterAlphabet())), ('>USVI/30/2016|zika|VI30|2016-08-07|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet())), ('>USVI/39/2016|zika|VI39|2016-11-09|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?', Seq('NNNNNNNNNNNNGACAGTTCGAGTTTGAAGCGAAAGCTAGCAACAGTATCAACA...NNN', SingleLetterAlphabet()))]\n"
     ]
    }
   ],
   "source": [
    "print len(new_glm_set_all_samples['usvi'])\n",
    "print len(new_glm_set_subsampled['usvi'])\n",
    "print new_glm_set_all_samples['usvi']\n",
    "print new_glm_set_subsampled['usvi']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      ">BRA/2016/FC_DQ62D2|zika|KY785455|2016-04-06|south_america|brazil|brazil|brazil|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785455|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">BRA/2016/FC_DQ28D1|zika|KY014317|2016-03-21|south_america|brazil|brazil|brazil|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014317|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">Brazil/2016/ZBRX102|zika|KY559018|2016-02-25|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559018|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "3\n",
      ">Brazil/2016/ZBRX127|zika|KY559021|2016-03-10|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559021|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "4\n",
      ">SPH2015|zika|KU321639|2015-03-XX|south_america|brazil|brazil|brazil|genbank|genome|Cunha et al|https://www.ncbi.nlm.nih.gov/nuccore/KU321639|First Complete Genome Sequence of Zika Virus (Flaviviridae, Flavivirus) from an Autochthonous Transmission in Brazil|Genome Announc 4 (2) (2016)|https://www.ncbi.nlm.nih.gov/pubmed/26941134\n",
      "5\n",
      ">Brazil/2016/ZBRC18|zika|KY558992|2016-01-06|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY558992|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "6\n",
      ">HS_2015_BA_01|zika|KX520666|2015-08-XX|south_america|brazil|brazil|brazil|genbank|genome|Teixeira et al|https://www.ncbi.nlm.nih.gov/nuccore/KX520666|Full coding sequence of the Zika Virus isolate HS-2015-BA-01|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "7\n",
      ">BRA/2016/FC_DQ131D1|zika|KY014296|2016-04-18|south_america|brazil|brazil|brazil|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014296|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "8\n",
      ">Brazil/2016/ZBRX128|zika|KY559022|2016-03-13|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559022|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "9\n",
      ">BRA/2016/FC_6863|zika|KY785450|2016-04-12|south_america|brazil|brazil|brazil|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785450|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      ">HND/2016/HU_ME180|zika|KY785431|2016-06-13|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785431|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">HND/2016/HU_ME38|zika|KY014319|2016-04-30|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014319|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">HND/2016/HU_SZ76|zika|KY785471|2016-05-03|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785471|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "3\n",
      ">FHT1175/HON/2016|zika|KY693677|2016-08-26|north_america|honduras|honduras|honduras|genbank|genome|Leguia et al|https://www.ncbi.nlm.nih.gov/nuccore/KY693677|Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "4\n",
      ">HND/2016/HU_ME171|zika|KY785448|2016-06-10|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785448|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "5\n",
      ">HND/2016/HU_ME136|zika|KY785461|2016-05-31|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785461|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "6\n",
      ">HND/2016/HU_ME167|zika|KY014327|2016-06-09|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014327|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">FHT1166/HON/2016|zika|KY693676|2016-08-26|north_america|honduras|honduras|honduras|genbank|genome|Leguia et al|https://www.ncbi.nlm.nih.gov/nuccore/KY693676|Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "8\n",
      ">HND/R103451/2015|zika|KX694534|2015-01-06|north_america|honduras|honduras|honduras|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX694534|Direct Submission|Submitted (12-AUG-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "9\n",
      ">HND/2016/HU_ME178|zika|KY014306|2016-06-10|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014306|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "0\n",
      ">DOM/2016/BB_0076|zika|KY014305|2016-04-05|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014305|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">DOM/2016/BB_0183|zika|KY785420|2016-04-18|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785420|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">DOM/2016/BB_0091|zika|KY785475|2016-04-07|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785475|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "3\n",
      ">DOM/2016/BB_0433|zika|KY785441|2016-06-13|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785441|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "4\n",
      ">R114916|zika|KX766028|2016-06-06|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Lanciotti et al|https://www.ncbi.nlm.nih.gov/nuccore/KX766028|Direct Submission|Submitted (23-AUG-2016) Diagnostic & Reference Laboratory, Arbovirus Diseases Branch, Centers for Disease Control & Prevention, 3150 Rampart Road, Fort Collins, CO 80521, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "5\n",
      ">DOM/2016/BB_0071|zika|KY785463|2016-04-05|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785463|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "6\n",
      ">DOM/2016/BB_0127|zika|KY014303|2016-04-11|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014303|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">DOM/2016/MA_WGS16_013|zika|KY785473|2016-06-15|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785473|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "8\n",
      ">DOM/2016/BB_0208|zika|KY014300|2016-04-20|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014300|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "9\n",
      ">DOM/2016/MA_WGS16_007|zika|KY785453|2016-05-10|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785453|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "0\n",
      ">Aedes_aegypti/USA/2016/FL01|zika|KY014324|2016-08-22|north_america|usa|usa|usa|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014324|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">USA/2016/FLUR006|zika|KY325467|2016-07-29|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325467|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "2\n",
      ">USA/2016/FL022|zika|KY075935|2016-07-22|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075935|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "3\n",
      ">USA/2016/FLUR063|zika|KY325477|2016-10-03|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325477|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "4\n",
      ">USA/2016/FL030|zika|KX922704|2016-08-02|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KX922704|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "5\n",
      ">USA/2016/FLUR013|zika|KY325472|2016-08-23|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325472|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "6\n",
      ">USA/2016/FL039|zika|KY014316|2016-08-17|north_america|usa|usa|usa|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014316|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">Aedes_aegypti/USA/2016/FL06|zika|KY075938|2016-09-20|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075938|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "8\n",
      ">USA/2016/FLUR002|zika|KY325466|2016-07-28|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325466|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "9\n",
      ">USA/2016/FL036|zika|KY075936|2016-08-04|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075936|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      ">COL/2016/SU_1856A|zika|KY785466|2016-04-28|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785466|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">COL/2016/SU_2293A|zika|KY785477|2016-06-09|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785477|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">Colombia/2016/ZC192Se|zika|KY317937|2016-01-07|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317937|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "3\n",
      ">Colombia/2016/ZC188Se|zika|KY317936|2016-01-16|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317936|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "4\n",
      ">COL/UF_1/2016|zika|KX247646|2016-02-09|south_america|colombia|colombia|colombia|genbank|genome|Lednicky et al|https://www.ncbi.nlm.nih.gov/nuccore/KX247646|Complete genomic sequence of Zika virus isolated from a traveler to Colombia in 2016|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "5\n",
      ">COL/2016/SU_2724A|zika|KY785469|2016-04-05|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785469|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "6\n",
      ">COL/2016/SU_1810A|zika|KY785417|2016-04-06|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785417|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">COL/FLR/2015|zika|KU820897|2015-12-XX|south_america|colombia|colombia|colombia|genbank|genome|Lahon et al|https://www.ncbi.nlm.nih.gov/nuccore/KU820897|Characterization of a Zika Virus Isolate from Colombia|PLoS Negl Trop Dis 10 (9), E0005019 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27654889\n",
      "8\n",
      ">Colombia/2016/ZC204Se|zika|KY317939|2016-01-06|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317939|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "9\n",
      ">Colombia/2016/ZC207Se|zika|KY317940|2016-01-09|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317940|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n",
      "0\n",
      ">USVI/6/2016|zika|VI6|2016-10-19|north_america|usvi|saint_john|saint_john|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "1\n",
      ">USVI/24/2016|zika|VI24|2016-09-13|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "2\n",
      ">USVI/32/2016|zika|VI32|2016-08-11|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "3\n",
      ">USVI/5/2016|zika|VI5|2016-10-17|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "4\n",
      ">USVI/11/2016|zika|VI11|2016-03-22|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "5\n",
      ">USVI/25/2016|zika|VI25|2016-09-27|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "6\n",
      ">USVI/36/2016|zika|VI36|2016-09-13|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "7\n",
      ">USVI/3/2016|zika|VI3|2016-09-26|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "8\n",
      ">USVI/30/2016|zika|VI30|2016-08-07|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "9\n",
      ">USVI/39/2016|zika|VI39|2016-11-09|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "0\n",
      "jamaica\n",
      "1\n",
      "honduras\n",
      "2\n",
      "dominican_republic\n",
      "3\n",
      "mexico\n",
      "4\n",
      "french_guiana\n",
      "5\n",
      "guatemala\n",
      "6\n",
      "colombia\n",
      "7\n",
      "martinique\n",
      "8\n",
      "haiti\n",
      "9\n",
      "nicaragua\n",
      "10\n",
      "brazil\n",
      "11\n",
      "suriname\n",
      "12\n",
      "usa\n",
      "13\n",
      "panama\n",
      "14\n",
      "cuba\n",
      "15\n",
      "el_salvador\n",
      "16\n",
      "puerto_rico\n",
      "17\n",
      "guadeloupe\n",
      "18\n",
      "peru\n",
      "19\n",
      "french_polynesia\n",
      "20\n",
      "usvi\n",
      "21\n",
      "venezuela\n",
      "22\n",
      "ecuador\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/min3_max10_seqs_augurFormat.fasta','w') as file:\n",
    "    for country in new_glm_set_subsampled.keys():\n",
    "    #print len(subsampled_demes[country])\n",
    "        for i in range(len(new_glm_set_subsampled[country])): #have a list of tuples (taxon,sequence)\n",
    "            header = new_glm_set_subsampled[country][i][0]#first element in the tuple ins the taxon name\n",
    "            file.write(str(header + '\\n' + str(new_glm_set_subsampled[country][i][1]) + '\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/min3_max10_seqs_augurFormat.fasta','w') as file:\n",
    "    for country in new_glm_set_subsampled.keys():\n",
    "    #print len(subsampled_demes[country])\n",
    "        for i in range(len(new_glm_set_subsampled[country])): #have a list of tuples (taxon,sequence)\n",
    "            header = subsampled_demes[country][i][0]#first element in the tuple ins the taxon name\n",
    "            file.write(str(header + '\\n' + subsampled_demes[country][i][1] + '\\n'))\n",
    "\n",
    "# with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/subsampled_seqs_max10.fasta','w') as file:\n",
    "#     for country in subsampled_demes.keys():\n",
    "#     #print len(subsampled_demes[country])\n",
    "#         for i in range(len(subsampled_demes[country])): #have a list of tuples (taxon,sequence)\n",
    "#             #print subsampled_demes[country][i][0]\n",
    "#             split_name = subsampled_demes[country][i][0].split('|') #first element in the tuple ins the taxon name\n",
    "#             #print split_name\n",
    "#             header = split_name[0] +'|'+ split_name[3] + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "#             file.write(str(header + '\\n' + subsampled_demes[country][i][1] + '\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">USVI/30/2016|zika|VI30|2016-08-07|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "['jamaica', 'honduras', 'mexico', 'french_guiana', 'colombia', 'nicaragua', 'brazil', 'peru', 'usa', 'panama', 'suriname', 'french_polynesia', 'venezuela', 'ecuador', 'dominican_republic', 'cuba', 'guatemala', 'martinique', 'haiti', 'guadeloupe', 'el_salvador', 'puerto_rico', 'usvi']\n",
      "23\n"
     ]
    }
   ],
   "source": [
    "subsampled_demes = {}\n",
    "for country in country_seq_dict.keys():\n",
    "    if len(country_seq_dict[country]) > 10:\n",
    "        subsampled_demes[country] = random.sample(country_seq_dict[country].items(), 10)  \n",
    "    else:\n",
    "        subsampled_demes[country] = [(key,country_seq_dict[country][key]) for key in country_seq_dict[country].keys()]\n",
    "        \n",
    "print subsampled_demes['usvi'][0][0]\n",
    "print subsampled_demes.keys()\n",
    "print len(subsampled_demes.keys())\n",
    "\n",
    "#for key in subsampled_demes.keys():\n",
    "    #print len(subsampled_demes[key])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "jamaica\n",
      "0\n",
      ">JAM/2016/MA_WGS16_039|zika|KY785430|2016-09-28|north_america|jamaica|jamaica|jamaica|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785430|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">JAM/2016/MA_WGS16_025|zika|KY785424|2016-07-10|north_america|jamaica|jamaica|jamaica|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785424|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">JAM/2016/MA_WGS16_041|zika|KY785432|2016-10-12|north_america|jamaica|jamaica|jamaica|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785432|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "honduras\n",
      "0\n",
      ">HND/2016/HU_ME136|zika|KY785461|2016-05-31|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785461|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">HND/2016/HU_ME131|zika|KY785414|2016-05-28|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785414|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">HND/2016/HU_ME152|zika|KY014315|2016-06-06|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014315|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "3\n",
      ">HND/2016/HU_ME147|zika|KY785442|2016-06-04|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785442|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "4\n",
      ">HND/2016/HU_ME50|zika|KY014311|2016-05-12|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014311|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "5\n",
      ">HND/2016/HU_SZ76|zika|KY785471|2016-05-03|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785471|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "6\n",
      ">HND/2016/HU_ME156|zika|KY785452|2016-06-07|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785452|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">HND/2016/HU_ME172|zika|KY785458|2016-06-10|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785458|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "8\n",
      ">HND/2016/HU_ME38|zika|KY014319|2016-04-30|north_america|honduras|honduras|honduras|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014319|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "9\n",
      ">HN16|zika|KY328289|2016-05-XX|north_america|honduras|honduras|honduras|genbank|genome|Murray et al|https://www.ncbi.nlm.nih.gov/nuccore/KY328289|Prolonged Detection of Zika Virus in Vaginal Secretions and Whole Blood|Emerging Infect. Dis. 23 (1), 99-101 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/27748649\n",
      "mexico\n",
      "0\n",
      ">Aedessp/MEX/MEX_I_7/2016|zika|KX446951|2016-01-01|north_america|mexico|mexico|mexico|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX446951|Direct Submission|Submitted (27-JUN-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "1\n",
      ">Aedes_aegypti/MEX/MEX_I_44/2016|zika|KY648934|2016-XX-XX|north_america|mexico|mexico|mexico|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY648934|Direct Submission|Submitted (22-FEB-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "2\n",
      ">MEX_CIENI551|zika|KY120348|2016-03-03|north_america|mexico|mexico|mexico|genbank|genome|Boukadida et al|https://www.ncbi.nlm.nih.gov/nuccore/KY120348|Complete Zika Virus Genome Sequences from a Serum Sample and after Isolation in Vero Cells|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "3\n",
      ">MEX/InDRE/2016|zika|KU922960|2016-02-25|north_america|mexico|mexico|mexico|genbank|genome|Diaz-Quinonez et al|https://www.ncbi.nlm.nih.gov/nuccore/KU922960|Molecular characterization of Zika virus in Mexico|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "4\n",
      ">USA/UT_1/2016|zika|KX827268|2016-01-01|north_america|mexico|mexico|mexico|genbank|genome|Swaminathan et al|https://www.ncbi.nlm.nih.gov/nuccore/KX827268|Fatal Zika Virus Infection with Secondary Nonsexual Transmission|N. Engl. J. Med. 375 (19), 1907-1909 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27681699\n",
      "5\n",
      ">R116265|zika|KX766029|2016-06-23|north_america|mexico|mexico|mexico|genbank|genome|Lanciotti et al|https://www.ncbi.nlm.nih.gov/nuccore/KX766029|Direct Submission|Submitted (23-AUG-2016) Diagnostic & Reference Laboratory, Arbovirus Diseases Branch, Centers for Disease Control & Prevention, 3150 Rampart Road, Fort Collins, CO 80521, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "6\n",
      ">Aedessp/MEX/MEX_2_81/2016|zika|KX446950|2016-01-01|north_america|mexico|mexico|mexico|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX446950|Direct Submission|Submitted (27-JUN-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "7\n",
      ">MEX_I_7|zika|KX247632|2015-11-XX|north_america|mexico|mexico|mexico|genbank|genome|Barrows et al|https://www.ncbi.nlm.nih.gov/nuccore/KX247632|Repurposed drug candidates to treat ZIKV infection in pregnancy|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "french_guiana\n",
      "0\n",
      ">V17271|zika|KU758877|2015-12-XX|south_america|french_guiana|french_guiana|french_guiana|genbank|genome|Enfissi et al|https://www.ncbi.nlm.nih.gov/nuccore/KU758877|Direct Submission|Submitted (25-FEB-2016) laboratoire de virologie, Institut Pasteur de la Guyane, 23 avenue Pasteur, Cayenne, French Guiana 97300, France|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "colombia\n",
      "0\n",
      ">COL/2016/SU_2724A|zika|KY785469|2016-04-05|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785469|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">COL/2016/SU_2293A|zika|KY785477|2016-06-09|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785477|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">Colombia/2016/ZC199Se|zika|KY317938|2016-01-10|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317938|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "3\n",
      ">Colombia/2016/ZC207Se|zika|KY317940|2016-01-09|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317940|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "4\n",
      ">COL/FLR/2015|zika|KU820897|2015-12-XX|south_america|colombia|colombia|colombia|genbank|genome|Lahon et al|https://www.ncbi.nlm.nih.gov/nuccore/KU820897|Characterization of a Zika Virus Isolate from Colombia|PLoS Negl Trop Dis 10 (9), E0005019 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27654889\n",
      "5\n",
      ">Colombia/2016/ZC192Se|zika|KY317937|2016-01-07|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317937|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "6\n",
      ">COL/UF_1/2016|zika|KX247646|2016-02-09|south_america|colombia|colombia|colombia|genbank|genome|Lednicky et al|https://www.ncbi.nlm.nih.gov/nuccore/KX247646|Complete genomic sequence of Zika virus isolated from a traveler to Colombia in 2016|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "7\n",
      ">Colombia/2016/ZC188Se|zika|KY317936|2016-01-16|south_america|colombia|colombia|colombia|genbank|genome|Quick et al|https://www.ncbi.nlm.nih.gov/nuccore/KY317936|Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples|Nat Protoc 12 (6), 1261-1276 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538739\n",
      "8\n",
      ">COL/2016/SU_1810A|zika|KY785417|2016-04-06|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785417|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "9\n",
      ">COL/2016/SU_1856A|zika|KY785466|2016-04-28|south_america|colombia|colombia|colombia|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785466|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "nicaragua\n",
      "0\n",
      ">NIC/5005_13A1/2016|zika|KY765327|2016-04-22|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY765327|Direct Submission|Submitted (13-MAR-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "1\n",
      ">NIC/4886_12A1/2016|zika|KY765318|2016-07-10|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY765318|Direct Submission|Submitted (13-MAR-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "2\n",
      ">NIC/7252_12A1/2016|zika|KY765317|2016-06-29|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY765317|Direct Submission|Submitted (13-MAR-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "3\n",
      ">Nica1_16|zika|KX421195|2016-01-19|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Tabata et al|https://www.ncbi.nlm.nih.gov/nuccore/KX421195|Zika Virus Targets Different Primary Human Placental Cells, Suggesting Two Routes for Vertical Transmission|Cell Host Microbe 20 (2), 155-166 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27443522\n",
      "4\n",
      ">NIC/6406_13A1/2016|zika|KY765320|2016-05-29|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY765320|Direct Submission|Submitted (13-MAR-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "5\n",
      ">NIC/8610_13A1/2016|zika|KY765324|2016-07-07|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY765324|Direct Submission|Submitted (13-MAR-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "6\n",
      ">Nica2_16|zika|KX421194|2016-01-13|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Tabata et al|https://www.ncbi.nlm.nih.gov/nuccore/KX421194|Zika Virus Targets Different Primary Human Placental Cells, Suggesting Two Routes for Vertical Transmission|Cell Host Microbe 20 (2), 155-166 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27443522\n",
      "7\n",
      ">NIC/6188_13A1/2016|zika|KY765323|2016-06-30|north_america|nicaragua|nicaragua|nicaragua|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KY765323|Direct Submission|Submitted (13-MAR-2017) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "brazil\n",
      "0\n",
      ">BRA/2016/FC_DQ62D2|zika|KY785455|2016-04-06|south_america|brazil|brazil|brazil|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785455|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">Brazil/2015/ZBRD107|zika|KY559002|2015-09-09|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559002|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "2\n",
      ">Brazil/2016/ZBRC14|zika|KY558990|2016-01-15|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY558990|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "3\n",
      ">Brazil/2016/ZBRX13|zika|KY559013|2016-04-24|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559013|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "4\n",
      ">Natal_RGN|zika|KU527068|2015-XX-XX|south_america|brazil|brazil|brazil|genbank|genome|Mlakar et al|https://www.ncbi.nlm.nih.gov/nuccore/KU527068|Zika Virus Associated with Microcephaly|N. Engl. J. Med. 374 (10), 951-958 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/26862926\n",
      "5\n",
      ">Brazil/2016/ZBRC18|zika|KY558992|2016-01-06|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY558992|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "6\n",
      ">BRA/2016/FC_DQ75D1|zika|KY785426|2016-04-04|south_america|brazil|brazil|brazil|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785426|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">Brazil/2016/ZBRE69|zika|KY559004|2016-04-16|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559004|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "8\n",
      ">Brazil/2016/ZBRX4|zika|KY559007|2016-04-18|south_america|brazil|brazil|brazil|genbank|genome|Faria et al|https://www.ncbi.nlm.nih.gov/nuccore/KY559007|Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "9\n",
      ">SSABR1|zika|KU707826|2015-07-01|south_america|brazil|brazil|brazil|genbank|genome|Giovanetti et al|https://www.ncbi.nlm.nih.gov/nuccore/KU707826|Zika virus complete genome from Salvador, Bahia, Brazil|Infect. Genet. Evol. 41, 142-145 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27071531\n",
      "peru\n",
      "0\n",
      ">FPI15263/PERU/Loreto/2016|zika|KY693679|2016-07-11|south_america|peru|peru|peru|genbank|genome|Leguia et al|https://www.ncbi.nlm.nih.gov/nuccore/KY693679|Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "1\n",
      ">FPI15198/PERU/Loreto/2016|zika|KY693678|2016-06-28|south_america|peru|peru|peru|genbank|genome|Leguia et al|https://www.ncbi.nlm.nih.gov/nuccore/KY693678|Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "usa\n",
      "0\n",
      ">USA/2016/FL028|zika|KY785443|2016-08-01|north_america|usa|usa|usa|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785443|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">USA/2016/FLUR058|zika|KY325476|2016-10-11|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325476|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "2\n",
      ">Aedes_aegypti/USA/2016/FL06|zika|KY075938|2016-09-20|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075938|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "3\n",
      ">USA/2016/FLUR005|zika|KY325481|2016-07-29|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325481|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "4\n",
      ">USA/2016/FLUR002|zika|KY325466|2016-07-28|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325466|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "5\n",
      ">USA/2016/FLWB042|zika|KY325478|2016-09-26|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325478|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "6\n",
      ">USA/2016/FLUR009|zika|KY325470|2016-08-04|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325470|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "7\n",
      ">USA/2016/FLUR026|zika|KY325474|2016-09-13|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325474|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "8\n",
      ">Aedes_aegypti/USA/2016/FL01|zika|KY014324|2016-08-22|north_america|usa|usa|usa|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014324|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "9\n",
      ">USA/2016/FLWB044|zika|KY325479|2016-09-28|north_america|usa|usa|usa|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY325479|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "panama\n",
      "0\n",
      ">PAN/CDC_259364_V1_V2/2015|zika|KX156776|2015-12-18|north_america|panama|panama|panama|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX156776|Direct Submission|Submitted (29-APR-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "1\n",
      ">PAN/CDC_259249_V1_V3/2015|zika|KX156775|2015-12-11|north_america|panama|panama|panama|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX156775|Direct Submission|Submitted (29-APR-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "2\n",
      ">PAN/CDC_259359_V1_V3/2015|zika|KX156774|2015-12-18|north_america|panama|panama|panama|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX156774|Direct Submission|Submitted (29-APR-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "3\n",
      ">PAN/BEI_259634_V4/2016|zika|KX198135|2016-XX-XX|north_america|panama|panama|panama|genbank|genome|Shabman et al|https://www.ncbi.nlm.nih.gov/nuccore/KX198135|Direct Submission|Submitted (05-MAY-2016) J. Craig Venter Institute, 9704 Medical Center Drive, Rockville, MD 20850, USA|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "suriname\n",
      "0\n",
      ">Z1106033|zika|KU312312|2015-10-02|south_america|suriname|suriname|suriname|genbank|genome|Enfissi et al|https://www.ncbi.nlm.nih.gov/nuccore/KU312312|Zika virus genome from the Americas|Lancet 387 (10015), 227-228 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/26775124\n",
      "1\n",
      ">SL1602|zika|KY348640|2016-01-22|south_america|suriname|suriname|suriname|genbank|genome|van Boheemen et al|https://www.ncbi.nlm.nih.gov/nuccore/KY348640|Quasispecies composition and evolution of a typical Zika virus clinical isolate from Suriname|Sci Rep 7 (1), 2368 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28539654\n",
      "french_polynesia\n",
      "0\n",
      ">PF13/251013_18|zika|KX369547|2013-10-25|oceania|french_polynesia|french_polynesia|french_polynesia|genbank|genome|Troesemeier et al|https://www.ncbi.nlm.nih.gov/nuccore/KX369547|Genome sequence of a candidate World Health Organization reference strain for Zika virus for nucleic acid testing|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "venezuela\n",
      "0\n",
      ">VEN/UF_1/2016|zika|KX702400|2016-03-25|south_america|venezuela|venezuela|venezuela|genbank|genome|Blohm et al|https://www.ncbi.nlm.nih.gov/nuccore/KX702400|Complete Genome Sequences of Identical Zika virus Isolates in a Nursing Mother and Her Infant|Genome Announc 5 (17), e00231-17 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28450510\n",
      "1\n",
      ">GZ02/2016|zika|KX056898|2016-02-25|south_america|venezuela|venezuela|venezuela|genbank|genome|Li et al|https://www.ncbi.nlm.nih.gov/nuccore/KX056898|The isolation and characterization of Zika virus imported into Guangdong, China|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "2\n",
      ">FVM00318/VEN/Maracay/2016|zika|KY693680|2016-10-19|south_america|venezuela|venezuela|venezuela|genbank|genome|Leguia et al|https://www.ncbi.nlm.nih.gov/nuccore/KY693680|Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "3\n",
      ">GZ01|zika|KU820898|2016-02-14|south_america|venezuela|venezuela|venezuela|genbank|genome|Deng et al|https://www.ncbi.nlm.nih.gov/nuccore/KU820898|Complete genome of Zika virus GZ01 isolate from China|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "4\n",
      ">Z16019|zika|KU955590|2016-02-26|south_america|venezuela|venezuela|venezuela|genbank|genome|Wu et al|https://www.ncbi.nlm.nih.gov/nuccore/KU955590|Large returning population of oversea Chinese of Guangdong province combined with tourists increase the risks of ZIKV transmission in Guangdong, China|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "ecuador\n",
      "0\n",
      ">EcEs089_16|zika|KX879604|2016-04-XX|south_america|ecuador|ecuador|ecuador|genbank|genome|Marquez et al|https://www.ncbi.nlm.nih.gov/nuccore/KX879604|First Complete Genome Sequences of Zika Virus Isolated from Febrile Patient Sera in Ecuador|Genome Announc 5 (8), e01673-16 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28232448\n",
      "1\n",
      ">EcEs062_16|zika|KX879603|2016-04-XX|south_america|ecuador|ecuador|ecuador|genbank|genome|Marquez et al|https://www.ncbi.nlm.nih.gov/nuccore/KX879603|First Complete Genome Sequences of Zika Virus Isolated from Febrile Patient Sera in Ecuador|Genome Announc 5 (8), e01673-16 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28232448\n",
      "dominican_republic\n",
      "0\n",
      ">DOM/2016/MA_WGS16_014|zika|KY785470|2016-06-18|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785470|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">DOM/2016/MA_WGS16_007|zika|KY785453|2016-05-10|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785453|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">DOM/2016/BB_0436|zika|KY014314|2016-06-14|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014314|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "3\n",
      ">DOM/2016/BB_0085|zika|KY785465|2016-04-07|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785465|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "4\n",
      ">DOM/2016/MA_WGS16_024|zika|KY785435|2016-07-07|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785435|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "5\n",
      ">DOM/2016/BB_0208|zika|KY014300|2016-04-20|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014300|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "6\n",
      ">DOM/2016/MA_WGS16_040|zika|KY785413|2016-10-06|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785413|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "7\n",
      ">DOM/2016/BB_0180|zika|KY014304|2016-04-18|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014304|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "8\n",
      ">DOM/2016/BB_0127|zika|KY014303|2016-04-11|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY014303|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "9\n",
      ">DOM/2016/MA_WGS16_013|zika|KY785473|2016-06-15|north_america|dominican_republic|dominican_republic|dominican_republic|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785473|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "cuba\n",
      "0\n",
      ">Cuba_2017|zika|MF438286|2017-02-12|north_america|cuba|cuba|cuba|genbank|genome|Lasala-Sanchez et al|https://www.ncbi.nlm.nih.gov/nuccore/MF438286|Zika virus polyprotein sequence isolated from a fetus-brain of Cuba pregnant traveler diagnosed in Spain (Madrid)|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "1\n",
      ">Cuba/2017/Hu0046Sa|zika|MF159531|2017-04-19|north_america|cuba|cuba|cuba|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/MF159531|Zika virus sequence from Cuba traveler diagnosed in the United States (Miami, FL)|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "guatemala\n",
      "0\n",
      ">V8375|zika|KU501217|2015-11-01|north_america|guatemala|guatemala|guatemala|genbank|genome|Lanciotti et al|https://www.ncbi.nlm.nih.gov/nuccore/KU501217|Phylogeny of Zika Virus in Western Hemisphere, 2015|Emerging Infect. Dis. 22 (5), 933-935 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27088323\n",
      "1\n",
      ">V103344|zika|KU501216|2015-12-01|north_america|guatemala|guatemala|guatemala|genbank|genome|Lanciotti et al|https://www.ncbi.nlm.nih.gov/nuccore/KU501216|Phylogeny of Zika Virus in Western Hemisphere, 2015|Emerging Infect. Dis. 22 (5), 933-935 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27088323\n",
      "2\n",
      ">FB_GWUH_2016|zika|KU870645|2016-02-02|north_america|guatemala|guatemala|guatemala|genbank|genome|Driggers et al|https://www.ncbi.nlm.nih.gov/nuccore/KU870645|Zika virus infection with prolonged maternal viremia and fetal brain abnormalities|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "martinique\n",
      "0\n",
      ">Martinique/2016/FL001|zika|KY785451|2016-03-22|north_america|martinique|martinique|martinique|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785451|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "1\n",
      ">MRS_OPY_Martinique_PaRi_2015|zika|KU647676|2015-12-XX|north_america|martinique|martinique|martinique|genbank|genome|Piorkowski et al|https://www.ncbi.nlm.nih.gov/nuccore/KU647676|Complete coding sequence of Zika virus from Martinique outbreak in 2015|New Microbes New Infect 11, 52-53 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27274849\n",
      "2\n",
      ">Martinique/2016/FL001Sa|zika|KY075932|2016-03-22|north_america|martinique|martinique|martinique|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075932|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "haiti\n",
      "0\n",
      ">Haiti/2016/PD|zika|KX269878|2016-02-01|north_america|haiti|haiti|haiti|genbank|genome|Barzon et al|https://www.ncbi.nlm.nih.gov/nuccore/KX269878|Infection dynamics in a traveller with persistent shedding of Zika virus RNA in semen for six months after returning from Haiti to Italy, January 2016|Euro Surveill. 21 (32) (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27542178\n",
      "1\n",
      ">Haiti/1/2016|zika|KX051563|2016-02-05|north_america|haiti|haiti|haiti|genbank|genome|Iovine et al|https://www.ncbi.nlm.nih.gov/nuccore/KX051563|Coinfection With Zika and Dengue-2 Viruses in a Traveler Returning From Haiti, 2016: Clinical Presentation and Genetic Analysis|Clin. Infect. Dis. (2016) In press|https://www.ncbi.nlm.nih.gov/pubmed/27694479\n",
      "2\n",
      ">Haiti/1225/2014|zika|KU509998|2014-12-12|north_america|haiti|haiti|haiti|genbank|genome|Lednicky et al|https://www.ncbi.nlm.nih.gov/nuccore/KU509998|Zika Virus Outbreak in Haiti in 2014: Molecular and Clinical Data|PLoS Negl Trop Dis 10 (4), E0004687 (2016)|https://www.ncbi.nlm.nih.gov/pubmed/27111294\n",
      "guadeloupe\n",
      "0\n",
      ">PHE_Guadeloupe|zika|KX673530|2016-04-21|north_america|guadeloupe|guadeloupe|guadeloupe|genbank|genome|Atkinson et al|https://www.ncbi.nlm.nih.gov/nuccore/KX673530|Complete genomic sequence of Zika virus isolated from a clinical semen sample|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "el_salvador\n",
      "0\n",
      ">UNK/2016/MA_WGS16_029|zika|KY785454|2016-08-05|north_america|el_salvador|el_salvador|el_salvador|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785454|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "puerto_rico\n",
      "0\n",
      ">PuertoRico/2016/FL016U|zika|KY075934|2016-07-03|north_america|puerto_rico|puerto_rico|puerto_rico|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075934|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "1\n",
      ">PRI/2016/MA_WGS16_005|zika|KY785481|2016-04-12|north_america|puerto_rico|puerto_rico|puerto_rico|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785481|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "2\n",
      ">PRI/2016/MA_WGS16_004|zika|KY785464|2016-04-13|north_america|puerto_rico|puerto_rico|puerto_rico|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785464|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "3\n",
      ">PuertoRico/2016/FL008U|zika|KY075933|2016-06-21|north_america|puerto_rico|puerto_rico|puerto_rico|genbank|genome|Grubaugh et al|https://www.ncbi.nlm.nih.gov/nuccore/KY075933|Genomic epidemiology reveals multiple introductions of Zika virus into the United States|Nature (2017) In press|https://www.ncbi.nlm.nih.gov/pubmed/28538723\n",
      "4\n",
      ">PRVABC59|zika|KX377337|2015-12-XX|north_america|puerto_rico|puerto_rico|puerto_rico|genbank|genome|Yun et al|https://www.ncbi.nlm.nih.gov/nuccore/KX377337|Complete Genome Sequences of Three Historically Important, Spatiotemporally Distinct, and Genetically Divergent Strains of Zika Virus: MR-766, P6-740, and PRVABC-59|Unpublished|https://www.ncbi.nlm.nih.gov/pubmed/\n",
      "5\n",
      ">PRI/2016/MA_WGS16_016|zika|KY785462|2016-06-26|north_america|puerto_rico|puerto_rico|puerto_rico|genbank|genome|Metsky et al|https://www.ncbi.nlm.nih.gov/nuccore/KY785462|Zika virus evolution and spread in the Americas|Nature 546 (7658), 411-415 (2017)|https://www.ncbi.nlm.nih.gov/pubmed/28538734\n",
      "usvi\n",
      "0\n",
      ">USVI/30/2016|zika|VI30|2016-08-07|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "1\n",
      ">USVI/7/2016|zika|VI7|2016-10-27|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "2\n",
      ">USVI/20/2016|zika|VI20|2016-10-13|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "3\n",
      ">USVI/23/2016|zika|VI23|2016-07-12|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "4\n",
      ">USVI/1/2016|zika|VI1|2016-09-28|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "5\n",
      ">USVI/38/2016|zika|VI38|2016-10-25|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "6\n",
      ">USVI/21/2016|zika|VI21|2016-11-07|north_america|usvi|saint_john|saint_john|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "7\n",
      ">USIV/35/2016|zika|VI35|2016-09-08|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "8\n",
      ">USVI/27/2016|zika|VI27|2016-08-19|north_america|usvi|saint_thomas|saint_thomas|fh|genome|Black et al|https://github.com/blab/zika-usvi/|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n",
      "9\n",
      ">USVI/12/2016|zika|VI12|2016-11-04|north_america|usvi|saint_croix|saint_croix|fh|genome|Black et al|https://github.com/blab/zika-usvi/tree/master/data|Genetic characterization of the Zika virus epidemic in the US Virgin Islands|?|?\n"
     ]
    }
   ],
   "source": [
    "#with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/subsampled_seqs_augurFormat.fasta','w') as file:\n",
    "for country in subsampled_demes.keys():\n",
    "    print country\n",
    "    for i in range(len(subsampled_demes[country])):\n",
    "        print i\n",
    "        taxon_name = subsampled_demes[country][i][0]\n",
    "        print taxon_name\n",
    "        sequence = subsampled_demes[country][i][1]\n",
    "        #print sequence\n",
    "            #file.write(str(taxon_name) + '\\n' + str(sequence) + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/subsampled_seqs_augurFormat.fasta','w') as file:\n",
    "    for country in subsampled_demes.keys():\n",
    "    #print len(subsampled_demes[country])\n",
    "        for i in range(len(subsampled_demes[country])): #have a list of tuples (taxon,sequence)\n",
    "            header = subsampled_demes[country][i][0]#first element in the tuple ins the taxon name\n",
    "            file.write(str(header + '\\n' + subsampled_demes[country][i][1] + '\\n'))\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/subsampled_seqs_max10.fasta','w') as file:\n",
    "    for country in subsampled_demes.keys():\n",
    "    #print len(subsampled_demes[country])\n",
    "        for i in range(len(subsampled_demes[country])): #have a list of tuples (taxon,sequence)\n",
    "            #print subsampled_demes[country][i][0]\n",
    "            split_name = subsampled_demes[country][i][0].split('|') #first element in the tuple ins the taxon name\n",
    "            #print split_name\n",
    "            header = split_name[0] +'|'+ split_name[3] + '|'+ split_name[4] + '|'+ split_name[5]\n",
    "            file.write(str(header + '\\n' + subsampled_demes[country][i][1] + '\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Aside:\n",
    "Below is the code I used to initally start looking at accessions and publications to determine which genomes were freely available to include in the analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# want dict where {publication: [accession, accession]}\n",
    "publication_accessions_dict = {}\n",
    "for pub in list_of_pubs:\n",
    "    accessions = []\n",
    "    for key in fauna_dict.keys():\n",
    "        taxa = fauna_dict[key].description\n",
    "        if taxa.split('|')[12] == pub:\n",
    "            accessions.append(taxa.split('|')[2])\n",
    "    publication_accessions_dict[pub] = accessions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pub_seq_count = 0\n",
    "for pub in published_submissions:\n",
    "    print len(publication_accessions_dict[pub])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "published_seqs = defaultdict(int)\n",
    "\n",
    "list_of_pubs = [strain_header_dict[key].split('|')[12] for key in zika_msa_dict.keys()]\n",
    "\n",
    "for pub in list_of_pubs:\n",
    "    published_seqs[pub] += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "published_submissions= ['First Complete Genome Sequence of Zika Virus (Flaviviridae, Flavivirus) from an Autochthonous Transmission in Brazil',\n",
    "'Genetic characterization of the Zika virus epidemic in the US Virgin Islands',\n",
    "'Isolation of Infective Zika Virus from Urine and Saliva of Patients in Brazil',\n",
    "'Full-length infectious cDNA clone of Zika virus from 2015 epidemic in Brazil: development and characterization of recombinant viruses in cell lines from human placenta, testis, and brain',\n",
    "'Zika virus complete genome from Salvador, Bahia, Brazil',\n",
    "'Complete Genome Sequences of Three Historically Important, Spatiotemporally Distinct, and Genetically Divergent Strains of Zika Virus: MR-766, P6-740, and PRVABC-59',\n",
    "'Isolation of infectious Zika virus from saliva and prolonged viral RNA shedding in a traveller returning from the Dominican Republic to Italy, January 2016',\n",
    "'Fatal Zika Virus Infection with Secondary Nonsexual Transmission',\n",
    "'Quasispecies composition and evolution of a typical Zika virus clinical isolate from Suriname',\n",
    "'First Complete Genome Sequences of Zika Virus Isolated from Febrile Patient Sera in Ecuador',\n",
    "'Epidemic establishment and cryptic transmission of Zika virus in Brazil and the Americas',\n",
    "'Genomic epidemiology reveals multiple introductions of Zika virus into the United States',\n",
    "'Coinfection With Zika and Dengue-2 Viruses in a Traveler Returning From Haiti, 2016: Clinical Presentation and Genetic Analysis',\n",
    "'Multiplex PCR method for MinION and Illumina sequencing of Zika and other virus genomes directly from clinical samples',\n",
    "'Large returning population of oversea Chinese of Guangdong province combined with tourists increase the risks of ZIKV transmission in Guangdong, China',\n",
    "'Zika Virus Targets Different Primary Human Placental Cells, Suggesting Two Routes for Vertical Transmission',\n",
    "'Establishment and cryptic transmission of Zika virus in Brazil and the Americas',\n",
    "'Zika Virus Outbreak in Haiti in 2014: Molecular and Clinical Data',\n",
    "'Prolonged Detection of Zika Virus in Vaginal Secretions and Whole Blood',\n",
    "'Complete Genome Sequences of Identical Zika virus Isolates in a Nursing Mother and Her Infant',\n",
    "'?',\n",
    "'Zika virus in the Americas: early epidemiological and genetic findings',\n",
    "'Full Genome Sequence and sfRNA Interferon Antagonist Activity of Zika Virus from Recife, Brazil',\n",
    "'Distinct Zika Virus Lineage in Salvador, Bahia, Brazil',\n",
    "'Genome sequence of a candidate World Health Organization reference strain for Zika virus for nucleic acid testing',\n",
    "'Zika virus evolution and spread in the Americas',\n",
    "'Zika virus genome from the Americas',\n",
    "'Complete coding sequence of Zika virus from Martinique outbreak in 2015',\n",
    "'Zika Virus Associated with Microcephaly',\n",
    "'Characterization of a Zika Virus Isolate from Colombia',\n",
    "'Detection and sequencing of Zika virus from amniotic fluid of fetuses with microcephaly in Brazil: a case study',\n",
    "'Phylogeny of Zika Virus in Western Hemisphere, 2015',\n",
    "'Complete genomic sequence of Zika virus isolated from a clinical semen sample',\n",
    "'Infection dynamics in a traveller with persistent shedding of Zika virus RNA in semen for six months after returning from Haiti to Italy, January 2016',                        \n",
    "                  ]\n",
    "\n",
    "published_submissions_with_incorrect_publication_name = [ 'Complete Zika Virus Genome Sequences from a Serum Sample and after Isolation in Vero Cells',\n",
    "'Full coding sequence of the Zika Virus isolate HS-2015-BA-01',\n",
    "'Large returning population of oversea Chinese of Guangdong province combined with tourists increase the risks of ZIKV transmission in Guangdong, China',\n",
    "'The isolation and characterization of Zika virus imported into Guangdong, China',\n",
    "'Full genomic characterization of a Zika virus isolate from Colombia',\n",
    "'Molecular characterization of Zika virus in Mexico',\n",
    "'Repurposed drug candidates to treat ZIKV infection in pregnancy',\n",
    "'Full-genome amplification and sequencing of Zika viruses using a targeted amplification approach',\n",
    "'Zika virus infection with prolonged maternal viremia and fetal brain abnormalities',\n",
    "'Complete genome of Zika virus GZ01 isolate from China',\n",
    "'Complete genomic sequence of Zika virus isolated from a traveler to Colombia in 2016',\n",
    "                                                       \n",
    "                                                         \n",
    "]\n",
    "\n",
    "unpublished_submissions = [\n",
    "'Imported cases of Zika virus disease in Russia in 2016-2017',\n",
    "'Complete Genome of a Zika Virus Isolate, Amazonas, Brazil, 2016',\n",
    "'Imported cases of Zika virus disease in Russia in 2016-2017',\n",
    "'Zika viral sequence identified in Fortaleza, Brazil',\n",
    "'Comparative analysis between the genomes of intra-host and cell cultured Zika virus obtained from a Mexican symptomatic patient'\n",
    "'Zika virus from breast milk',\n",
    "'Genome sequences of Zika viruses isolated in Nicaragua',              \n",
    "'Emergence of Zika virus in Brazil: Detection of Asian genotype',\n",
    "'Brazilian Zika virus genome sequence',\n",
    "'Zika viruses in Haiti during an outbreak of Chikungunya Fever in mid 2014', #white et al sequences\n",
    "'Zika virus sequence from Cuba traveler diagnosed in the United States (Miami, FL)', #nate\n",
    "'Zika virus genomes from human cases in Florida, USA',\n",
    "'Isolation of infectious Zika virus from sera in a traveller returning from Brazil to Italy',\n",
    "'Zika virus infection in pregnant mice',\n",
    "'Zika virus polyprotein sequence isolated from a fetus-brain of Cuba pregnant traveler diagnosed in Spain (Madrid)',\n",
    "'Genome sequences of four Zika virus isolates from Brazil',\n",
    "'Direct Submission'\n",
    "                          ]\n",
    "\n",
    "unpublished_seqs_to_pursue = [\n",
    "                              'Genome sequences of Zika viruses isolated in Nicaragua',\n",
    "                              'Imported cases of Zika virus disease in Russia in 2016-2017', #dominican republic and mexico\n",
    "                              'Zika viruses in Haiti during an outbreak of Chikungunya Fever in mid 2014',\n",
    "                              'Zika virus polyprotein sequence isolated from a fetus-brain of Cuba pregnant traveler diagnosed in Spain (Madrid)', #cuba\n",
    "                              'Genome sequences of four Zika virus isolates from Brazil' #brazil early...published on, but not by the submitting authors... \n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "published_fauna_seqs = {key:value for key,value in fauna_dict.items() if fauna_dict[key].description.split('|')[12] in published_submissions}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/published-sequences-in-analysis.tsv','w') as file:\n",
    "    file.write('{}\\t{}\\t{}\\n'.format('strain_name', 'accession_number', 'lead_author'))\n",
    "    for key in published_fauna_seqs.keys():\n",
    "        file.write('{}\\t{}\\t{}\\n'.format(key.split('|')[0], key.split('|')[2], key.split('|')[-1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'published_fauna_seqs' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-0e231e70935f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mprint\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpublished_fauna_seqs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/augur-published-seqs-only.fasta'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mpublished_fauna_seqs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mpublished_fauna_seqs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdescription\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'|'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mregions_to_exclude1\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m#don't write out anything except American seqs:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'published_fauna_seqs' is not defined"
     ]
    }
   ],
   "source": [
    "print len(published_fauna_seqs)\n",
    "\n",
    "with open('/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/augur-published-seqs-only.fasta','w') as file:\n",
    "    for key in published_fauna_seqs.keys():\n",
    "        if published_fauna_seqs[key].description.split('|')[4] in regions_to_exclude1: #don't write out anything except American seqs:\n",
    "            continue\n",
    "        else: \n",
    "            file.write(str('>' + published_fauna_seqs[key].description + '\\n' + published_fauna_seqs[key].seq + '\\n'))\n",
    "    file.write(str('>' + published_fauna_seqs['PF13/251013_18|zika|KX369547|2013-10-25|oceania|french_polynesia|french_polynesia|french_polynesia|genbank|genome|Troesemeier'].description+ '\\n' + published_fauna_seqs[key].seq + '\\n'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'output_dict' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-b7cd81f95ece>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/Users/alliblk/Desktop/gitrepos/zika-usvi/data/fastas/usvi_primary_clade.fasta'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mout_file\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;32min\u001b[0m \u001b[0moutput_dict\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'|'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32min\u001b[0m \u001b[0musvi_primary_clade\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m             \u001b[0;32mprint\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'output_dict' is not defined"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
